// @hitachivantara/uikit-react-core/dist/cjs/FilterGroup/RightPanel/RightPanel.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const CounterLabel = require("../../utils/CounterLabel.cjs");
const setId = require("../../utils/setId.cjs");
const FilterGroupContext = require("../FilterGroupContext.cjs");
const RightPanel_styles = require("./RightPanel.styles.cjs");
const CheckBox = require("../../CheckBox/CheckBox.cjs");
const Panel = require("../../Panel/Panel.cjs");
const Input = require("../../Input/Input.cjs");
const List = require("../../List/List.cjs");
const HvFilterGroupRightPanel = ({
  id,
  className,
  labels,
  emptyElement,
  classes: classesProp
}) => {
  const { classes } = RightPanel_styles.useClasses(classesProp);
  const [searchStr, setSearchStr] = React.useState("");
  const [allSelected, setAllSelected] = React.useState(false);
  const [anySelected, setAnySelected] = React.useState(false);
  const {
    filterOptions,
    filterValues = [],
    setFilterValues,
    activeGroup
  } = React.useContext(FilterGroupContext.HvFilterGroupContext);
  const { all: allActiveGroupOptions, enabled: enabledActiveGroupOptions } = React.useMemo(() => {
    const filteredOptions = filterOptions[activeGroup]?.data.filter(
      (option) => option.name.toLowerCase().includes(searchStr.toLowerCase())
    );
    return {
      all: filteredOptions?.map((option) => option.id) || [],
      enabled: filteredOptions?.filter((option) => !option.disabled)?.map((option) => option.id) || []
    };
  }, [filterOptions, activeGroup, searchStr]);
  const activeFilterValues = React.useMemo(
    () => filterValues[activeGroup]?.filter(
      (value) => allActiveGroupOptions.includes(value)
    ) || [],
    [filterValues, allActiveGroupOptions, activeGroup]
  );
  const listValues = React.useMemo(
    () => filterOptions[activeGroup]?.data.map((option) => ({
      ...option,
      label: option.name,
      selected: filterValues[activeGroup]?.includes(option.id),
      isHidden: option.name.toLowerCase().indexOf(searchStr.toLowerCase()) < 0
    })) || [],
    [filterOptions, filterValues, activeGroup, searchStr]
  );
  const updateSelectAll = React.useCallback(() => {
    const nbrSelected = activeFilterValues?.length;
    const hasSelection = nbrSelected > 0;
    const allSelect = nbrSelected === allActiveGroupOptions.length;
    setAnySelected(hasSelection);
    setAllSelected(hasSelection && allSelect);
  }, [activeFilterValues, allActiveGroupOptions]);
  React.useEffect(() => {
    updateSelectAll();
  }, [activeFilterValues, updateSelectAll]);
  React.useEffect(() => setSearchStr(""), [activeGroup]);
  const onChangeHandler = (values) => {
    const newFilterValues = filterOptions.map(
      (_, i) => activeGroup === i ? values.filter((v) => v.selected).map((v) => v.id) : [...filterValues[i] || []]
    );
    setFilterValues(newFilterValues);
  };
  const handleSelectAll = React.useCallback(() => {
    const newFilterValues = structuredClone(filterValues);
    if (anySelected) {
      if (searchStr !== "") {
        newFilterValues[activeGroup] = filterValues[activeGroup]?.filter(
          (value) => !enabledActiveGroupOptions.includes(value)
        );
      } else {
        newFilterValues[activeGroup] = [];
      }
    } else {
      const currentOptions = newFilterValues[activeGroup] || [];
      newFilterValues[activeGroup] = [
        ...currentOptions,
        ...enabledActiveGroupOptions
      ];
    }
    setFilterValues(newFilterValues);
  }, [
    activeGroup,
    enabledActiveGroupOptions,
    anySelected,
    filterValues,
    setFilterValues,
    searchStr
  ]);
  const SelectAll = React.useCallback(() => {
    return /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.selectAllContainer, children: /* @__PURE__ */ jsxRuntime.jsx(
      CheckBox.HvCheckBox,
      {
        id: setId.setId(id, "select-all"),
        label: /* @__PURE__ */ jsxRuntime.jsx(
          CounterLabel.CounterLabel,
          {
            selected: activeFilterValues?.length,
            total: allActiveGroupOptions.length,
            conjunctionLabel: labels?.multiSelectionConjunction
          }
        ),
        onChange: () => handleSelectAll(),
        className: classes.selectAll,
        indeterminate: anySelected && !allSelected,
        checked: allSelected
      }
    ) });
  }, [
    activeFilterValues?.length,
    allActiveGroupOptions.length,
    allSelected,
    anySelected,
    handleSelectAll,
    id,
    labels,
    classes?.selectAllContainer,
    classes?.selectAll
  ]);
  return /* @__PURE__ */ jsxRuntime.jsx(Panel.HvPanel, { id: setId.setId(id, "rightPanel"), className, children: listValues.length > 0 ? /* @__PURE__ */ jsxRuntime.jsxs(jsxRuntime.Fragment, { children: [
    /* @__PURE__ */ jsxRuntime.jsx(
      Input.HvInput,
      {
        id: setId.setId(id, "search"),
        classes: {
          root: classes.search
        },
        type: "search",
        placeholder: labels?.searchBoxPlaceholder,
        value: searchStr,
        onChange: (_, str) => setSearchStr(str)
      }
    ),
    /* @__PURE__ */ jsxRuntime.jsx(SelectAll, {}),
    /* @__PURE__ */ jsxRuntime.jsx(
      List.HvList,
      {
        id: setId.setId(id, "list"),
        values: listValues,
        className: classes.list,
        multiSelect: true,
        useSelector: true,
        showSelectAll: false,
        onChange: onChangeHandler,
        selectable: true,
        condensed: true,
        hasTooltips: true
      },
      activeGroup
    )
  ] }) : emptyElement });
};
exports.filterGroupRightPanelClasses = RightPanel_styles.staticClasses;
exports.HvFilterGroupRightPanel = HvFilterGroupRightPanel;
