// @hitachivantara/uikit-react-core/dist/esm/ColorPicker/ColorPicker.js
import { jsxs, jsx, Fragment } from "react/jsx-runtime";
import { forwardRef } from "react";
import { ColorPicker } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useControlled } from "../hooks/useControlled.js";
import { useLabels } from "../hooks/useLabels.js";
import { useUniqueId } from "../hooks/useUniqueId.js";
import { setId } from "../utils/setId.js";
import { useClasses } from "./ColorPicker.styles.js";
import { staticClasses } from "./ColorPicker.styles.js";
import { PresetColors } from "./PresetColors/PresetColors.js";
import { Picker } from "./Picker/Picker.js";
import { SavedColors } from "./SavedColors/SavedColors.js";
import { HvFormElement } from "../FormElement/FormElement.js";
import { HvLabel } from "../FormElement/Label/Label.js";
import { HvInfoMessage } from "../FormElement/InfoMessage/InfoMessage.js";
import { HvBaseDropdown } from "../BaseDropdown/BaseDropdown.js";
import { HvPanel } from "../Panel/Panel.js";
import { HvTypography } from "../Typography/Typography.js";
const DEFAULT_LABELS = {
  recommendedColorsLabel: "Recommended colors:",
  customColorsLabel: "Custom colors:"
};
const HvColorPicker = forwardRef(
  function HvColorPicker2(props, ref) {
    const {
      id,
      name,
      required,
      disabled,
      label,
      "aria-label": ariaLabel,
      "aria-labelledby": ariaLabelledBy,
      description,
      "aria-describedby": ariaDescribedBy,
      className,
      classes: classesProp,
      value,
      onChange,
      onChangeComplete,
      defaultValue = "",
      expanded,
      defaultExpanded = false,
      recommendedColorsPosition = "top",
      recommendedColors = [
        "#95AFE8",
        "#E89E5D",
        "#83B8A6",
        "#70759C",
        "#C57E7E",
        "#FADA95",
        "#ADBFE8",
        "#E3B386",
        "#9AC6B7",
        "#8B90AF",
        "#CF9797",
        "#FAE1AA"
      ],
      showLabels = true,
      labels: labelsProp,
      dropdownIcon = "colorPicker",
      placeholder = "Select color...",
      iconOnly = false,
      showSavedColors = true,
      showCustomColors = true,
      savedColorsValue,
      defaultSavedColorsValue = [],
      onSavedColorAdded,
      onSavedColorRemoved,
      deleteSavedColorButtonArialLabel = "Delete saved color",
      addSavedColorButtonAriaLabel = "Add current color to saved colors"
    } = useDefaultProps("HvColorPicker", props);
    const { classes, cx } = useClasses(classesProp);
    const labels = useLabels(DEFAULT_LABELS, labelsProp);
    const [isOpen, setIsOpen] = useControlled(expanded, defaultExpanded);
    const [color, setColor] = useControlled(value, defaultValue);
    const [savedColors, setSavedColors] = useControlled(
      savedColorsValue,
      defaultSavedColorsValue
    );
    const elementId = useUniqueId(id);
    const hasLabel = label != null;
    const hasDescription = description != null;
    const handleToggle = (_, open) => {
      setIsOpen(open);
    };
    const handleSelect = (val) => {
      onChange?.(val.hex);
      onChangeComplete?.(val.hex);
      setColor(val.hex);
    };
    const handleOnChange = (val) => {
      onChange?.(val.hex);
      setColor(val.hex);
    };
    const handleOnChangeComplete = (val) => {
      onChangeComplete?.(val.hex);
      setColor(val.hex);
    };
    const handleAddColor = () => {
      const colorToAdd = color || "#000000";
      onSavedColorAdded?.(colorToAdd);
      setSavedColors([...savedColors, colorToAdd]);
    };
    const handleRemoveColor = (val, position) => {
      if (savedColors[position] === val) {
        const sColors = [...savedColors];
        sColors.splice(position, 1);
        onSavedColorRemoved?.(val);
        setSavedColors(sColors);
      }
    };
    const setFocusToContent = (containerRef) => {
      const inputs = containerRef?.getElementsByTagName("input");
      if (inputs && inputs.length > 0) {
        inputs[0].focus();
      }
    };
    return /* @__PURE__ */ jsxs(
      HvFormElement,
      {
        id,
        name,
        disabled,
        required,
        className: cx(classes.root, className),
        children: [
          (hasLabel || hasDescription) && /* @__PURE__ */ jsxs("div", { className: classes.labelContainer, children: [
            hasLabel && /* @__PURE__ */ jsx(
              HvLabel,
              {
                showGutter: true,
                id: setId(elementId, "label"),
                label,
                className: classes.label
              }
            ),
            hasDescription && /* @__PURE__ */ jsx(
              HvInfoMessage,
              {
                id: setId(elementId, "description"),
                className: classes.description,
                children: description
              }
            )
          ] }),
          /* @__PURE__ */ jsx(
            HvBaseDropdown,
            {
              ref,
              variableWidth: true,
              className,
              expanded: isOpen,
              onToggle: handleToggle,
              onContainerCreation: setFocusToContent,
              classes: {
                root: cx({ [classes.dropdownRootIconOnly]: iconOnly })
              },
              adornment: iconOnly && color ? /* @__PURE__ */ jsx(
                "div",
                {
                  style: { backgroundColor: color },
                  className: cx(
                    classes.headerColorIcon,
                    classes.headerColorIconOnly
                  )
                }
              ) : dropdownIcon === "colorPicker" ? /* @__PURE__ */ jsx(ColorPicker, { className: classes.colorPickerIcon }) : void 0,
              placeholder: iconOnly ? void 0 : color ? /* @__PURE__ */ jsxs(Fragment, { children: [
                /* @__PURE__ */ jsx(
                  "div",
                  {
                    style: { backgroundColor: color },
                    className: classes.headerColorIcon
                  }
                ),
                /* @__PURE__ */ jsx(
                  HvTypography,
                  {
                    className: classes.headerColorValue,
                    variant: "label",
                    children: color
                  }
                )
              ] }) : placeholder,
              "aria-label": ariaLabel,
              "aria-labelledby": [label && setId(elementId, "label"), ariaLabelledBy].join(" ").trim() || void 0,
              "aria-describedby": [description && setId(elementId, "description"), ariaDescribedBy].join(" ").trim() || void 0,
              children: /* @__PURE__ */ jsx(HvPanel, { className: classes.panel, children: /* @__PURE__ */ jsxs("div", { className: classes.colorPicker, children: [
                recommendedColorsPosition === "top" && /* @__PURE__ */ jsx(
                  PresetColors,
                  {
                    className: classes.recommendedColorsRoot,
                    colors: recommendedColors,
                    onClick: handleSelect,
                    title: showLabels ? labels.recommendedColorsLabel : void 0
                  }
                ),
                showCustomColors && /* @__PURE__ */ jsx(
                  Picker,
                  {
                    classes: {
                      fields: cx({
                        [classes.pickerFields]: recommendedColorsPosition === "bottom" || showSavedColors
                      })
                    },
                    title: showLabels ? labels.customColorsLabel : void 0,
                    color,
                    onChange: handleOnChange,
                    onChangeComplete: handleOnChangeComplete
                  }
                ),
                showSavedColors && /* @__PURE__ */ jsx(
                  SavedColors,
                  {
                    colors: savedColors,
                    onAddColor: handleAddColor,
                    onClickColor: handleSelect,
                    onRemoveColor: handleRemoveColor,
                    deleteButtonAriaLabel: deleteSavedColorButtonArialLabel,
                    addButtonAriaLabel: addSavedColorButtonAriaLabel
                  }
                ),
                recommendedColorsPosition === "bottom" && /* @__PURE__ */ jsx(
                  PresetColors,
                  {
                    colors: recommendedColors,
                    onClick: handleSelect,
                    title: showLabels ? labels.recommendedColorsLabel : void 0
                  }
                )
              ] }) })
            }
          )
        ]
      }
    );
  }
);
export {
  HvColorPicker,
  staticClasses as colorPickerClasses
};
