// @hitachivantara/uikit-styles/dist/cjs/index.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const ds3 = require("./themes/ds3.cjs");
const ds5 = require("./themes/ds5.cjs");
const pentahoPlus = require("./themes/pentahoPlus.cjs");
const theme = require("./theme.cjs");
const makeTheme = require("./makeTheme.cjs");
const utils = require("./utils.cjs");
const CssBaseline = require("./CssBaseline.cjs");
const breakpoints = require("./tokens/breakpoints.cjs");
const colors = require("./tokens/colors.cjs");
const radii = require("./tokens/radii.cjs");
const space = require("./tokens/space.cjs");
const sizes = require("./tokens/sizes.cjs");
const typography = require("./tokens/typography.cjs");
const zIndices = require("./tokens/zIndices.cjs");
const themes = { ds3: ds3.default, ds5: ds5.default, pentahoPlus: pentahoPlus.default };
exports.ds3 = ds3.default;
exports.ds5 = ds5.default;
exports.pentahoPlus = pentahoPlus.default;
exports.getColor = theme.getColor;
exports.theme = theme.theme;
exports.makeTheme = makeTheme.makeTheme;
exports.getThemesList = utils.getThemesList;
exports.getThemesVars = utils.getThemesVars;
exports.hasMultipleArgs = utils.hasMultipleArgs;
exports.mapCSSVars = utils.mapCSSVars;
exports.mergeTheme = utils.mergeTheme;
exports.parseTheme = utils.parseTheme;
exports.spacingUtil = utils.spacingUtil;
exports.spacingUtilOld = utils.spacingUtilOld;
exports.CssBaseline = CssBaseline.CssBaseline;
exports.CssScopedBaseline = CssBaseline.CssScopedBaseline;
exports.breakpoints = breakpoints.breakpoints;
exports.colors = colors.colors;
exports.radii = radii.radii;
exports.space = space.space;
exports.sizes = sizes.sizes;
exports.fontFamily = typography.fontFamily;
exports.fontSizes = typography.fontSizes;
exports.fontWeights = typography.fontWeights;
exports.lineHeights = typography.lineHeights;
exports.zIndices = zIndices.zIndices;
exports.themes = themes;
