// @hitachivantara/uikit-react-core/dist/cjs/AppSwitcher/AppSwitcher.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvAppSwitcher", {
  root: {
    display: "flex",
    flexDirection: "column",
    overflow: "hidden"
  },
  item: {},
  itemSelected: {},
  itemDisabled: {},
  itemTrigger: {},
  itemIcon: {},
  itemTitle: {},
  itemInfoIcon: {},
  actionsContainer: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "flex-start",
    overflowY: "auto",
    gap: uikitStyles.theme.space.xs,
    padding: 4,
    margin: -4
  },
  footerContainer: {
    display: "flex",
    alignItems: "center",
    marginTop: "auto",
    height: 52,
    paddingTop: uikitStyles.theme.space.sm
  },
  open: {
    zIndex: uikitStyles.theme.zIndices.overlay,
    position: "absolute",
    top: "50px",
    overflowX: "hidden",
    boxShadow: uikitStyles.theme.colors.shadow
  },
  closed: { display: "none" },
  title: {
    minHeight: 36,
    paddingBottom: uikitStyles.theme.space.sm,
    ...uikitStyles.theme.typography.label
  },
  titleAnchor: {
    WebkitLineClamp: 2
  },
  single: { width: 280 + 40 },
  dual: { width: 560 + 40 },
  fluid: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
