// @hitachivantara/uikit-react-core/dist/cjs/Typography/Typography.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvTypography", {
  root: {
    fontFamily: uikitStyles.theme.fontFamily.body
  },
  disabled: {
    color: uikitStyles.theme.colors.secondary_60
  },
  isLink: {
    cursor: "pointer",
    color: uikitStyles.theme.colors.primary,
    textDecoration: "underline"
  },
  noWrap: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  },
  // variants
  display: {},
  title1: {},
  title2: {},
  title3: {},
  title4: {},
  body: {},
  label: {},
  captionLabel: {},
  caption1: {},
  caption2: {},
  // legacy variants
  // TODO: remove (legacy or all variants) in v6 in favour of dynamic variants
  "5xlTitle": {},
  "4xlTitle": {},
  // @ts-ignore non-existent variant
  "3xlTitle": {},
  xxlTitle: {},
  xlTitle: {},
  lTitle: {},
  mTitle: {},
  sTitle: {},
  xsTitle: {},
  xxsTitle: {},
  sectionTitle: {
    textTransform: "uppercase"
  },
  highlightText: {},
  normalText: {},
  placeholderText: {},
  link: {
    cursor: "pointer",
    textDecoration: "underline"
  },
  disabledText: {},
  selectedNavText: {},
  vizText: {},
  vizTextDisabled: {},
  xsInlineLink: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
