// @hitachivantara/uikit-react-core/dist/esm/Banner/BannerContent/BannerContent.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { forwardRef } from "react";
import SnackbarContent from "@mui/material/SnackbarContent";
import { Close } from "@hitachivantara/uikit-react-icons";
import { iconVariant } from "../../utils/iconVariant.js";
import { setId } from "../../utils/setId.js";
import { useClasses } from "./BannerContent.styles.js";
import { staticClasses } from "./BannerContent.styles.js";
import { HvButton } from "../../Button/Button.js";
import { HvActionsGeneric } from "../../ActionsGeneric/ActionsGeneric.js";
const HvBannerContent = forwardRef(
  function HvBannerContent2(props, ref) {
    const {
      id,
      classes: classesProp,
      className,
      showIcon = false,
      customIcon,
      variant = "default",
      onClose,
      actions,
      actionsCallback,
      // TODO - remove in v6
      onAction,
      actionsPosition = "auto",
      content,
      children,
      actionProps,
      ...others
    } = props;
    const { classes, cx } = useClasses(classesProp);
    const icon = customIcon || showIcon && iconVariant(variant, "base_dark");
    const effectiveActionsPosition = actionsPosition === "auto" ? "inline" : actionsPosition;
    const handleAction = (evt, action) => {
      onAction?.(evt, action);
      actionsCallback?.(evt, id, action);
    };
    return /* @__PURE__ */ jsx("div", { className: classes.outContainer, children: /* @__PURE__ */ jsx(
      SnackbarContent,
      {
        ref,
        id,
        classes: {
          root: classes.root,
          message: classes.message,
          action: classes.action
        },
        className: cx(classes.baseVariant, classes[variant], className),
        message: /* @__PURE__ */ jsxs(Fragment, { children: [
          icon && /* @__PURE__ */ jsx("div", { className: classes.iconContainer, children: icon }),
          /* @__PURE__ */ jsx(
            "div",
            {
              id: setId(id, "message-text"),
              className: classes.messageContainer,
              children: children ?? content
            }
          ),
          actions && effectiveActionsPosition === "inline" && /* @__PURE__ */ jsx(
            "div",
            {
              id: setId(id, "message-actions"),
              className: classes.messageActions,
              children: /* @__PURE__ */ jsx(
                HvActionsGeneric,
                {
                  id,
                  variant: "semantic",
                  actions,
                  onAction: handleAction
                }
              )
            }
          )
        ] }),
        action: /* @__PURE__ */ jsxs("div", { className: classes.actionContainer, children: [
          /* @__PURE__ */ jsx(
            HvButton,
            {
              icon: true,
              className: classes.closeAction,
              variant: "semantic",
              "aria-label": "Close",
              onClick: onClose,
              ...actionProps,
              children: /* @__PURE__ */ jsx(Close, { size: "XS" })
            }
          ),
          actions && effectiveActionsPosition === "bottom-right" && /* @__PURE__ */ jsx("div", { className: classes.actionsInnerContainer, children: /* @__PURE__ */ jsx(
            HvActionsGeneric,
            {
              id,
              variant: "semantic",
              actions,
              onAction: handleAction
            }
          ) })
        ] }),
        ...others
      }
    ) });
  }
);
export {
  HvBannerContent,
  staticClasses as bannerContentClasses
};
