// @hitachivantara/uikit-react-core/dist/cjs/DropDownMenu/DropDownMenu.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvDropDownMenu", {
  container: {},
  baseContainer: {
    "&[data-popper-placement=bottom-end] .HvBaseDropdown-panel": {
      borderRadius: `${uikitStyles.theme.radii.base} 0 ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base}`
    },
    "&[data-popper-placement=bottom-start] .HvBaseDropdown-panel": {
      borderRadius: `0 ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base}`
    },
    "&[data-popper-placement=top-start] .HvBaseDropdown-panel": {
      borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0`
    },
    "&[data-popper-placement=top-end] .HvBaseDropdown-panel": {
      borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0 ${uikitStyles.theme.radii.base}`
    }
  },
  root: {
    display: "inline-block",
    width: "auto"
  },
  icon: {},
  iconSelected: {
    boxShadow: uikitStyles.theme.colors.shadow
  },
  menuListRoot: {
    padding: uikitStyles.theme.space.sm
  },
  menuList: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
