// @hitachivantara/uikit-react-core/dist/cjs/Tabs/Tab/Tab.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const Tab = require("@mui/material/Tab");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const Tab_styles = require("./Tab.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Tab__default = /* @__PURE__ */ _interopDefault(Tab);
const HvTab = React.forwardRef(function HvTab2(props, ref) {
  const {
    classes: classesProp,
    iconPosition = "top",
    disabled = false,
    ...others
  } = uikitReactUtils.useDefaultProps("HvTab", props);
  const { classes, cx } = Tab_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tab__default.default,
    {
      ref,
      classes: {
        root: classes.root,
        selected: classes.selected,
        disabled: classes.disabled
      },
      focusVisibleClassName: cx("HvIsFocusVisible", classes.focusVisible),
      disabled,
      iconPosition,
      ...others
    }
  );
});
exports.tabClasses = Tab_styles.staticClasses;
exports.HvTab = HvTab;
