// @hitachivantara/uikit-react-core/dist/cjs/Tabs/Tab/Tab.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvTab", {
  root: {
    marginTop: "3px",
    padding: uikitStyles.theme.spacing(0, "sm"),
    minWidth: 70,
    minHeight: 32,
    textTransform: "none",
    ...uikitStyles.theme.typography.body,
    borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0 0`,
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
      "&::after": {
        height: "1px",
        backgroundColor: uikitStyles.theme.colors.atmo4
      }
    },
    "&$selected": {
      fontWeight: uikitStyles.theme.typography.label.fontWeight
    },
    "&$disabled": {
      color: uikitStyles.theme.colors.secondary_60,
      cursor: "not-allowed",
      pointerEvents: "all",
      opacity: 1,
      "&:hover": {
        background: "none"
      }
    },
    opacity: 1,
    "&::after": {
      position: "absolute",
      left: 0,
      top: "calc(100% - 1px)",
      height: "1px",
      width: "100%",
      backgroundColor: uikitStyles.theme.colors.atmo4,
      content: "''"
    },
    // Override Mui styling: https://mui.com/material-ui/api/tab/#css
    "& .MuiTab-iconWrapper": {
      margin: 0
    },
    "& svg .color0": {
      fill: "currentcolor"
    }
  },
  focusVisible: {
    ...focusUtils.outlineStyles
  },
  selected: {},
  disabled: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
