// @hitachivantara/uikit-react-core/dist/cjs/Dialog/Title/Title.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const MuiDialogTitle = require("@mui/material/DialogTitle");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const iconVariant = require("../../utils/iconVariant.cjs");
const context = require("../context.cjs");
const Title_styles = require("./Title.styles.cjs");
const Typography = require("../../Typography/Typography.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiDialogTitle__default = /* @__PURE__ */ _interopDefault(MuiDialogTitle);
const HvDialogTitle = (props) => {
  const {
    classes: classesProp,
    className,
    children,
    variant = "default",
    showIcon = true,
    customIcon,
    ...others
  } = uikitReactUtils.useDefaultProps("HvDialogTitle", props);
  const { classes, cx } = Title_styles.useClasses(classesProp);
  const { fullScreen } = context.useDialogContext();
  const icon = customIcon || showIcon && iconVariant.iconVariant(variant);
  return /* @__PURE__ */ jsxRuntime.jsxs(
    Typography.HvTypography,
    {
      component: MuiDialogTitle__default.default,
      variant: "title4",
      className: cx(
        classes.root,
        classes.messageContainer,
        {
          [classes.fullscreen]: fullScreen,
          [classes.textWithIcon]: icon,
          [classes.titleText]: typeof children === "string"
        },
        className
      ),
      ...others,
      children: [
        icon,
        children
      ]
    }
  );
};
exports.dialogTitleClasses = Title_styles.staticClasses;
exports.HvDialogTitle = HvDialogTitle;
