// @hitachivantara/uikit-react-core/dist/cjs/Skeleton/Skeleton.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const Skeleton_styles = require("./Skeleton.styles.cjs");
const HvSkeleton = (props) => {
  const {
    classes: classesProp,
    className,
    children,
    hidden = false,
    variant = "text",
    animation = "pulse",
    backgroundImage,
    width,
    height,
    style,
    ...others
  } = uikitReactUtils.useDefaultProps("HvSkeleton", props);
  const { classes, cx } = Skeleton_styles.useClasses(classesProp);
  if (hidden) return /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.content, children });
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cx(
        classes.root,
        classes[variant],
        animation && classes[animation],
        className
      ),
      style: {
        width,
        height,
        ...backgroundImage && {
          backgroundImage: `url(${backgroundImage})`,
          backgroundSize: "contain",
          backgroundPosition: "center",
          backgroundRepeat: "no-repeat"
        },
        ...style
      },
      ...others,
      children
    }
  );
};
exports.skeletonClasses = Skeleton_styles.staticClasses;
exports.HvSkeleton = HvSkeleton;
