// @hitachivantara/uikit-react-core/dist/cjs/DatePicker/DatePicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvDatePicker", {
  root: {
    position: "relative"
  },
  actionContainer: {
    justifyContent: "space-between"
  },
  leftContainer: {},
  rightContainer: {},
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {},
  description: {},
  error: {},
  dropdown: {
    display: "block"
  },
  panel: {},
  dropdownHeaderInvalid: {
    border: `1px solid ${uikitStyles.theme.colors.negative_120}`,
    "&:hover": {
      border: `1px solid ${uikitStyles.theme.colors.negative_120}`
    }
  },
  dropdownHeaderOpen: {},
  icon: {},
  action: {
    "&:first-of-type": {
      marginRight: uikitStyles.theme.space.xs
    }
  },
  inputText: {},
  dateText: {
    color: "inherit",
    fontWeight: uikitStyles.theme.typography.label.fontWeight
  },
  container: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
