// @hitachivantara/uikit-react-core/dist/cjs/AppSwitcher/Action/Action.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses(
  "HvAppSwitcher-Action",
  {
    root: {
      width: "100%",
      maxWidth: 280,
      minHeight: 52
    },
    icon: { display: "flex", minWidth: 40, justifyContent: "center" },
    iconUrl: { width: 32 },
    iconInfo: { minWidth: 32 },
    disabled: {},
    selected: {},
    typography: {
      display: "flex",
      justifyContent: "flex-start",
      alignItems: "center",
      width: "100%",
      minHeight: 52,
      padding: `6px ${uikitStyles.theme.space.xs}`,
      border: "none",
      borderLeft: `solid 2px ${uikitStyles.theme.colors.secondary}`,
      cursor: "pointer",
      textDecoration: "inherit",
      color: "inherit",
      backgroundColor: "inherit",
      "$disabled &": {
        cursor: "not-allowed"
      }
    },
    title: {
      flexGrow: 1,
      whiteSpace: "normal",
      textAlign: "left",
      margin: `0 ${uikitStyles.theme.space.xs}`,
      textWrap: "balance",
      ...uikitStyles.theme.typography.label
    },
    titleAnchor: {
      WebkitLineClamp: 2
    }
  }
);
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
