// @hitachivantara/uikit-react-core/dist/cjs/Header/Header.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const Header_styles = require("./Header.styles.cjs");
const HvHeader = React.forwardRef(function HvHeader2(props, ref) {
  const {
    className,
    classes: classesProp,
    children,
    position = "fixed",
    ...others
  } = uikitReactUtils.useDefaultProps("HvHeader", props);
  const { classes, cx } = Header_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    "header",
    {
      ref,
      style: { position },
      className: cx(
        classes.root,
        classes.backgroundColor,
        { [classes.fixed]: position === "fixed" },
        className
      ),
      ...others,
      children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.header, children })
    }
  );
});
exports.headerClasses = Header_styles.staticClasses;
exports.HvHeader = HvHeader;
