// @hitachivantara/uikit-react-core/dist/cjs/Header/Header.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvHeader", {
  root: {
    height: uikitStyles.theme.header.height,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    display: "flex",
    flexDirection: "column",
    width: "100%",
    boxSizing: "border-box",
    flexShrink: 0,
    zIndex: uikitStyles.theme.zIndices.banner,
    borderTop: "none",
    boxShadow: uikitStyles.theme.colors.shadow
  },
  header: {
    display: "flex",
    alignItems: "center",
    width: "100%",
    height: "100%",
    padding: `0 ${uikitStyles.theme.space.sm}`,
    "& > *:not(nav)": {
      zIndex: 2
    }
  },
  fixed: { position: "fixed", top: 0, left: "auto", right: 0 },
  backgroundColor: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
