// @hitachivantara/uikit-react-core/dist/cjs/Dialog/Dialog.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvDialog", {
  root: {},
  background: { background: uikitStyles.theme.alpha("atmo4", 0.8) },
  paper: {
    color: uikitStyles.theme.colors.secondary,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    boxShadow: uikitStyles.theme.colors.shadow,
    borderColor: uikitStyles.theme.colors.atmo4,
    borderRadius: uikitStyles.theme.radii.round
  },
  fullscreen: {},
  fullHeight: {
    height: "100%"
  },
  closeButton: {
    padding: 0,
    minWidth: "auto",
    position: "absolute",
    top: uikitStyles.theme.space.sm,
    right: uikitStyles.theme.space.sm,
    width: 32,
    height: 32
  },
  statusBar: {
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
    borderTopWidth: 4
  },
  success: {
    borderColor: uikitStyles.theme.colors.positive
  },
  error: {
    borderColor: uikitStyles.theme.colors.negative
  },
  warning: {
    borderColor: uikitStyles.theme.colors.warning
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
