// @hitachivantara/uikit-uno-preset/dist/esm/theme.mjs
import { ds5 } from "@hitachivantara/uikit-styles";
const { dawn: defaultColors } = ds5.colors.modes;
const { base, ...hvSpacing } = ds5.space;
const hvBreakpoints = Object.entries(ds5.breakpoints.values).map(
  ([key, value]) => [key, `${value}px`]
);
const hvZIndex = Object.entries(ds5.zIndices).map(
  ([key, value]) => [key, `${value}`]
);
const extendTheme = (baseTheme) => ({
  ...baseTheme,
  borderRadius: {
    DEFAULT: ds5.radii.base,
    ...ds5.radii
  },
  breakpoints: Object.fromEntries(hvBreakpoints),
  containers: Object.fromEntries(
    hvBreakpoints.map(([k, v]) => [k, `(min-width: ${v})`])
  ),
  spacing: {
    DEFAULT: hvSpacing.xs,
    ...hvSpacing
  },
  zIndex: Object.fromEntries(hvZIndex),
  // colors
  colors: {
    ...baseTheme.colors,
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000000",
    white: "#ffffff",
    // using `theme` CSS vars for automatic theme switching, losing alpha + no HvProvider support
    ...defaultColors
  },
  accentColor: { DEFAULT: defaultColors.primary },
  textColor: { DEFAULT: defaultColors.secondary },
  backgroundColor: { DEFAULT: defaultColors.backgroundColor },
  // typography
  fontFamily: {
    body: ds5.fontFamily.body,
    sans: ds5.fontFamily.body,
    serif: ds5.fontFamily.body,
    mono: "monospace"
  },
  lineHeight: { DEFAULT: ds5.lineHeights.base, ...ds5.lineHeights },
  fontSize: { DEFAULT: ds5.fontSizes.base, ...ds5.fontSizes },
  fontWeight: { DEFAULT: ds5.fontWeights.normal, ...ds5.fontWeights }
});
const themeModes = {
  light: {
    colors: {
      ...ds5.colors.modes.dawn
    }
  },
  dark: {
    colors: {
      ...ds5.colors.modes.wicked
    }
  }
};
export {
  extendTheme,
  themeModes
};
