// @hitachivantara/uikit-react-core/dist/cjs/Tabs/Tabs.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const Tabs = require("@mui/material/Tabs");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const Tabs_styles = require("./Tabs.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Tabs__default = /* @__PURE__ */ _interopDefault(Tabs);
const HvTabs = React.forwardRef(function HvTabs2(props, ref) {
  const {
    classes: classesProp,
    floating,
    ...others
  } = uikitReactUtils.useDefaultProps("HvTabs", props);
  const { classes, cx } = Tabs_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tabs__default.default,
    {
      ref,
      classes: {
        root: cx(classes.root, { [classes.floating]: floating }),
        flexContainer: classes.flexContainer,
        indicator: classes.indicator,
        scroller: classes.scroller
      },
      ...others
    }
  );
});
exports.tabsClasses = Tabs_styles.staticClasses;
exports.HvTabs = HvTabs;
