// @hitachivantara/uikit-react-core/dist/cjs/Calendar/CalendarNavigation/MonthSelector/MonthSelector.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const utils = require("../../utils.cjs");
const MonthSelector_styles = require("./MonthSelector.styles.cjs");
const ButtonBase = require("../../../ButtonBase/ButtonBase.cjs");
const HvMonthSelector = ({
  classes: classesProp,
  id,
  locale,
  onChange,
  onViewModeChange,
  rangeMode,
  visibleMonth,
  ...others
}) => {
  const { classes, cx } = MonthSelector_styles.useClasses(classesProp);
  const listMonthNamesShort = utils.getMonthNamesList(locale, "short");
  return /* @__PURE__ */ jsxRuntime.jsx(
    "div",
    {
      className: cx(classes.calendarMonthlyGrid, {
        [classes.rangeModeWidth]: rangeMode,
        [classes.normalWidth]: !rangeMode
      }),
      children: listMonthNamesShort.map((monthName, index) => /* @__PURE__ */ jsxRuntime.jsx(
        ButtonBase.HvButtonBase,
        {
          className: cx(classes.calendarMonthlyCell, {
            [classes.calendarMonthlyCellSelected]: index + 1 === visibleMonth
          }),
          onClick: (event) => {
            onChange?.(event, "month", index + 1);
            onViewModeChange("calendar");
          },
          ...others,
          children: monthName
        },
        monthName
      ))
    }
  );
};
exports.monthSelectorClasses = MonthSelector_styles.staticClasses;
exports.HvMonthSelector = HvMonthSelector;
