// @hitachivantara/uikit-styles/dist/cjs/themes/pentahoPlus.cjs
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const makeTheme = require("../makeTheme.cjs");
const colors = require("../tokens/colors.cjs");
const colorsPalette = require("../tokens/colorsPalette.cjs");
const radii = require("../tokens/radii.cjs");
const ld = (c1, c2) => `light-dark(${c1}, ${c2})`;
const buttonColors = {
  primary: {
    subtleBg: ld(colorsPalette.blue[50], colorsPalette.blue[950]),
    subtleBorder: ld(colorsPalette.blue[200], colorsPalette.blue[800])
  },
  secondary: {
    subtleBg: ld(colorsPalette.slate[100], colorsPalette.slate[800]),
    subtleBorder: ld(colorsPalette.slate[300], colorsPalette.slate[700])
  },
  success: {
    subtleBorder: ld(colorsPalette.green[200], colorsPalette.green[800]),
    subtleBg: ld(colorsPalette.emerald[100], colorsPalette.green[900])
  },
  warning: {
    subtleBorder: ld(colorsPalette.amber[200], colorsPalette.amber[800]),
    subtleBg: ld(colorsPalette.amber[100], colorsPalette.amber[900])
  },
  error: {
    subtleBorder: ld(colorsPalette.red[200], colorsPalette.red[800]),
    subtleBg: ld(colorsPalette.red[100], colorsPalette.red[900])
  }
};
const inputColors = {
  bg: ld("white", "black"),
  border: ld(colorsPalette.slate[400], colorsPalette.slate[600]),
  borderHover: ld(colorsPalette.blue[600], colorsPalette.blue[600]),
  borderActive: ld(colorsPalette.slate[700], colorsPalette.slate[300])
};
const pentahoPlus = makeTheme.makeTheme((theme) => ({
  name: "pentahoPlus",
  colors: {
    modes: {
      dawn: {
        type: "light",
        ...colors.colors.common,
        ...colors.colors.light,
        brand: colorsPalette.blue[600],
        containerBackgroundHover: `color-mix(in srgb, ${colorsPalette.blue[600]} 10%, transparent)`,
        backgroundColor: colorsPalette.slate[100],
        atmo1: colorsPalette.slate[50],
        atmo2: colorsPalette.slate[100],
        atmo3: colorsPalette.slate[200],
        atmo4: colorsPalette.slate[300],
        base_light: colorsPalette.slate[50],
        base_dark: colorsPalette.slate[700],
        primary: colorsPalette.blue[600],
        primary_80: colorsPalette.blue[500],
        primary_20: `color-mix(in srgb, ${colorsPalette.blue[600]} 10%, transparent)`,
        secondary: colorsPalette.slate[700],
        secondary_80: colorsPalette.slate[500],
        secondary_60: colorsPalette.slate[400],
        secondary_20: `color-mix(in srgb, ${colorsPalette.slate[700]} 10%, transparent)`,
        // 🆕
        positive: colorsPalette.green[600],
        positive_120: colorsPalette.green[700],
        positive_80: colorsPalette.green[400],
        neutral: colorsPalette.sky[500],
        warning: colorsPalette.amber[500],
        warning_120: colorsPalette.orange[500],
        warning_140: colorsPalette.orange[700],
        negative: colorsPalette.red[600],
        negative_120: colorsPalette.red[700],
        negative_80: colorsPalette.red[400],
        catastrophic: colorsPalette.rose[800],
        negative_20: colorsPalette.red[100],
        warning_20: colorsPalette.amber[100],
        positive_20: colorsPalette.green[100],
        neutral_20: colorsPalette.sky[100],
        shadow: `0px 2px 4px -1px color-mix(in srgb, ${colorsPalette.slate[700]} 8%, transparent)`,
        shad1: `color-mix(in srgb, ${colorsPalette.slate[700]} 8%, transparent)`,
        pp: {
          primary: colorsPalette.blue[600],
          primaryAction: colorsPalette.blue[700],
          primaryStrong: colorsPalette.blue[800],
          primarySubtle: colorsPalette.blue[200],
          primaryDimmed: colorsPalette.blue[100],
          success: colorsPalette.green[600],
          successAction: colorsPalette.green[700],
          successStrong: colorsPalette.green[800],
          successDimmed: colorsPalette.green[50],
          warning: colorsPalette.amber[500],
          warningAction: colorsPalette.amber[600],
          warningStrong: colorsPalette.amber[700],
          warningDimmed: colorsPalette.amber[50],
          error: colorsPalette.red[600],
          errorAction: colorsPalette.red[700],
          errorStrong: colorsPalette.red[800],
          errorDimmed: colorsPalette.red[50],
          neutral: colorsPalette.cyan[500],
          neutralAction: colorsPalette.cyan[600],
          neutralStrong: colorsPalette.cyan[700],
          neutralDimmed: colorsPalette.cyan[50],
          text: colorsPalette.slate[700],
          textSubtle: colorsPalette.slate[500],
          textDisabled: colorsPalette.neutral[400],
          textDimmed: colorsPalette.slate[300],
          link: colorsPalette.blue[600],
          linkActive: colorsPalette.blue[700],
          divider: colorsPalette.slate[300],
          dividerSubtle: colorsPalette.slate[200],
          dividerDimmed: colorsPalette.slate[400],
          bgPage: colorsPalette.slate[100],
          bgSurface: colorsPalette.slate[50],
          bgActive: colorsPalette.slate[200],
          bgHover: colorsPalette.blue[100],
          bgDisabled: colorsPalette.neutral[200],
          bgOverlay: `color-mix(in srgb, ${colorsPalette.slate[900]} 60%, transparent)`,
          dimmer: "#FFFFFF",
          borderDisabled: colorsPalette.neutral[400]
        }
      },
      wicked: {
        type: "dark",
        ...colors.colors.common,
        ...colors.colors.dark,
        brand: colorsPalette.blue[600],
        containerBackgroundHover: `color-mix(in srgb, ${colorsPalette.blue[500]} 10%, transparent)`,
        backgroundColor: colorsPalette.slate[900],
        atmo1: colorsPalette.slate[800],
        atmo2: colorsPalette.slate[900],
        atmo3: colorsPalette.slate[950],
        atmo4: colorsPalette.slate[700],
        base_light: colorsPalette.slate[50],
        base_dark: colorsPalette.slate[700],
        primary: colorsPalette.blue[500],
        primary_80: colorsPalette.blue[400],
        primary_20: `color-mix(in srgb, ${colorsPalette.blue[500]} 10%, transparent)`,
        secondary: colorsPalette.slate[50],
        secondary_80: colorsPalette.slate[200],
        secondary_60: colorsPalette.slate[300],
        secondary_20: `color-mix(in srgb, ${colorsPalette.slate[50]} 10%, transparent)`,
        // 🆕
        positive: colorsPalette.green[500],
        positive_120: colorsPalette.green[600],
        positive_80: colorsPalette.green[400],
        neutral: colorsPalette.sky[500],
        warning: colorsPalette.amber[500],
        warning_120: colorsPalette.amber[600],
        warning_140: colorsPalette.orange[600],
        negative: colorsPalette.red[500],
        negative_120: colorsPalette.red[600],
        negative_80: colorsPalette.red[300],
        catastrophic: colorsPalette.pink[600],
        negative_20: colorsPalette.red[100],
        warning_20: colorsPalette.amber[100],
        positive_20: colorsPalette.green[100],
        neutral_20: colorsPalette.sky[100],
        shadow: `0px 2px 4px -1px color-mix(in srgb, ${colorsPalette.slate[700]} 8%, transparent)`,
        shad1: `color-mix(in srgb, ${colorsPalette.slate[700]} 8%, transparent)`,
        pp: {
          primary: colorsPalette.blue[400],
          primaryAction: colorsPalette.blue[300],
          primaryStrong: colorsPalette.blue[200],
          primarySubtle: colorsPalette.blue[900],
          primaryDimmed: colorsPalette.blue[950],
          success: colorsPalette.green[600],
          successAction: colorsPalette.green[500],
          successStrong: colorsPalette.green[300],
          successDimmed: colorsPalette.green[950],
          warning: colorsPalette.yellow[500],
          warningAction: colorsPalette.yellow[400],
          warningStrong: colorsPalette.yellow[300],
          warningDimmed: colorsPalette.yellow[950],
          error: colorsPalette.red[600],
          errorAction: colorsPalette.red[500],
          errorStrong: colorsPalette.red[300],
          errorDimmed: colorsPalette.red[950],
          neutral: colorsPalette.cyan[500],
          neutralAction: colorsPalette.cyan[400],
          neutralStrong: colorsPalette.cyan[300],
          neutralDimmed: colorsPalette.cyan[950],
          text: colorsPalette.slate[50],
          textSubtle: colorsPalette.slate[400],
          textDisabled: colorsPalette.neutral[500],
          textDimmed: colorsPalette.slate[700],
          link: colorsPalette.blue[400],
          linkActive: colorsPalette.blue[300],
          divider: colorsPalette.slate[700],
          dividerSubtle: colorsPalette.slate[700],
          dividerDimmed: colorsPalette.slate[500],
          bgPage: colorsPalette.slate[950],
          bgSurface: colorsPalette.slate[900],
          bgActive: colorsPalette.slate[900],
          bgHover: colorsPalette.blue[950],
          bgDisabled: colorsPalette.neutral[900],
          bgOverlay: `color-mix(in srgb, ${colorsPalette.slate[900]} 40%, transparent)`,
          dimmer: "#000000",
          borderDisabled: colorsPalette.neutral[700]
        }
      }
    }
  },
  fontFamily: {
    body: "Inter, Arial, Helvetica, sans-serif"
  },
  typography: {
    display: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.xl4,
      lineHeight: theme.lineHeights.xl4,
      letterSpacing: "0.00504em"
    },
    title1: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.xl3,
      lineHeight: theme.lineHeights.xl3,
      letterSpacing: "0.00384em"
    },
    title2: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.xl2,
      lineHeight: theme.lineHeights.xl2,
      letterSpacing: "0.00288em"
    },
    title3: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.xl,
      lineHeight: theme.lineHeights.xl,
      letterSpacing: "0.0024em"
    },
    title4: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.lg,
      lineHeight: theme.lineHeights.lg,
      letterSpacing: "0.00192em"
    },
    label: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.base,
      lineHeight: theme.lineHeights.base,
      letterSpacing: "0.00168em"
    },
    body: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.normal,
      fontSize: theme.fontSizes.base,
      lineHeight: theme.lineHeights.base,
      letterSpacing: "0.00168em"
    },
    captionLabel: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.semibold,
      fontSize: theme.fontSizes.sm,
      lineHeight: theme.lineHeights.sm,
      letterSpacing: 0
    },
    caption1: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.normal,
      fontSize: theme.fontSizes.sm,
      lineHeight: theme.lineHeights.sm,
      letterSpacing: "0.00144em"
    },
    caption2: {
      color: theme.colors.secondary,
      fontWeight: theme.fontWeights.normal,
      fontSize: theme.fontSizes.xs,
      lineHeight: theme.lineHeights.xs,
      letterSpacing: "0.0012em"
    },
    // LEGACY
    "5xlTitle": {
      color: theme.colors.secondary,
      fontSize: "52px",
      letterSpacing: "0.02em",
      lineHeight: "60px",
      fontWeight: 600
    },
    "4xlTitle": {
      color: theme.colors.secondary,
      fontSize: "52px",
      letterSpacing: "0.02em",
      lineHeight: "60px",
      fontWeight: 400
    },
    xxlTitle: {
      color: theme.colors.secondary,
      fontSize: "42px",
      letterSpacing: "0.02em",
      lineHeight: "52px",
      fontWeight: 400
    },
    lTitle: {
      color: theme.colors.secondary,
      fontSize: "32px",
      letterSpacing: "0.02em",
      lineHeight: "40px",
      fontWeight: 400
    },
    sTitle: {
      color: theme.colors.secondary,
      fontSize: "22px",
      letterSpacing: "0.02em",
      lineHeight: "30px",
      fontWeight: 400
    },
    xxsTitle: {
      color: theme.colors.secondary,
      fontSize: "18px",
      letterSpacing: "0.02em",
      lineHeight: "28px",
      fontWeight: 400
    },
    sectionTitle: {
      color: theme.colors.secondary,
      fontSize: "14px",
      letterSpacing: "0.32em",
      lineHeight: "18px",
      fontWeight: 400,
      textTransform: "uppercase"
    },
    placeholderText: {
      color: theme.colors.secondary_60,
      fontSize: "12px",
      letterSpacing: "0.02em",
      lineHeight: "16px",
      fontWeight: 400
    },
    link: {
      color: theme.colors.primary,
      fontSize: "12px",
      letterSpacing: "0.02em",
      lineHeight: "16px",
      fontWeight: 600
    },
    disabledText: {
      color: theme.colors.secondary_60,
      fontSize: "12px",
      letterSpacing: "0.02em",
      lineHeight: "16px",
      fontWeight: 600
    },
    selectedNavText: {
      color: theme.colors.brand,
      fontSize: "12px",
      letterSpacing: "0.02em",
      lineHeight: "16px",
      fontWeight: 600
    },
    vizTextDisabled: {
      color: theme.colors.secondary_60,
      fontSize: "10px",
      letterSpacing: "0.02em",
      lineHeight: "15px",
      fontWeight: 400
    },
    xsInlineLink: {
      color: theme.colors.primary,
      fontSize: "10px",
      letterSpacing: "0.02em",
      lineHeight: "15px",
      fontWeight: 600,
      textDecoration: "underline"
    }
  },
  sizes: {
    xs: "32px",
    md: "48px",
    lg: "56px",
    xl: "64px",
    sm: "40px"
  },
  radii: {
    ...radii.radii,
    base: "6px"
  },
  components: {
    HvLoading: {
      classes: {
        loadingBar: {
          borderRadius: 1
        }
      }
    },
    HvBadge: {
      classes: {
        badgePosition: {
          color: theme.colors.base_light,
          backgroundColor: "#334155"
        }
      }
    },
    HvBaseCheckBox: {
      classes: {
        root: {
          "& svg": {
            width: 16,
            height: 16,
            borderRadius: "3px",
            border: `1px solid ${theme.colors.pp.textSubtle}`
          },
          "&:hover": {
            backgroundColor: theme.colors.pp.bgHover
          }
        },
        checked: {
          "& svg": {
            border: `1px solid ${theme.colors.primary}`,
            backgroundColor: theme.colors.primary,
            color: theme.colors.atmo1
          },
          "&.HvBaseCheckBox-disabled": {
            "& svg": {
              color: theme.colors.pp.textDisabled
            }
          }
        },
        disabled: {
          "&.HvBaseCheckBox-root": {
            "& svg": {
              border: `1px solid ${theme.colors.pp.textDisabled}`,
              backgroundColor: theme.colors.pp.bgDisabled
            }
          }
        },
        indeterminate: {
          "& svg": {
            backgroundColor: theme.colors.pp.bgSurface,
            border: `1px solid ${theme.colors.pp.textSubtle}`,
            color: theme.colors.pp.textSubtle
          },
          "&.HvBaseCheckBox-disabled": {
            "& svg": {
              color: theme.colors.pp.textDisabled
            }
          }
        }
      }
    },
    HvBaseRadio: {
      classes: {
        root: {
          "& svg": {
            backgroundColor: theme.colors.pp.bgSurface,
            borderColor: theme.colors.pp.textSubtle
          },
          "&:hover": {
            backgroundColor: theme.colors.pp.bgHover
          }
        },
        checked: {
          "& svg": {
            borderColor: theme.colors.primary,
            backgroundColor: theme.colors.primary
          },
          "&.HvBaseRadio-disabled": {
            "& svg": {
              borderColor: theme.colors.pp.textDisabled,
              backgroundColor: theme.colors.pp.textDisabled,
              color: theme.colors.pp.bgDisabled
            }
          }
        },
        disabled: {
          "& svg": {
            border: `1px solid ${theme.colors.pp.textDisabled}`,
            backgroundColor: theme.colors.pp.bgDisabled
          }
        }
      }
    },
    HvDotPagination: {
      classes: {
        radio: {
          width: 16,
          minWidth: 16
        }
      }
    },
    HvBaseSwitch: {
      classes: {
        root: {
          "&:hover": {
            backgroundColor: theme.colors.pp.bgHover
          },
          "& .HvBaseSwitch-switchBase": {
            "& .HvBaseSwitch-thumb": {
              backgroundColor: theme.colors.pp.bgSurface,
              borderColor: theme.colors.pp.bgSurface,
              boxShadow: `0px 1px 1px 0px color-mix(in srgb, ${colorsPalette.slate[700]} 16%, transparent)`
            },
            "+.HvBaseSwitch-track": {
              height: 14,
              backgroundColor: theme.colors.pp.textDimmed,
              borderColor: theme.colors.pp.textSubtle
            },
            "&.HvBaseSwitch-checked": {
              "& .HvBaseSwitch-thumb": {
                border: `1px solid ${theme.colors.pp.bgSurface}`
              },
              "+.HvBaseSwitch-track": {
                backgroundColor: theme.colors.pp.primary,
                borderColor: theme.colors.pp.primaryStrong
              }
            },
            "&.HvBaseSwitch-disabled": {
              "& .HvBaseSwitch-thumb": {
                backgroundColor: theme.colors.pp.borderDisabled,
                borderColor: theme.colors.pp.borderDisabled
              },
              "+.HvBaseSwitch-track": {
                borderColor: theme.colors.pp.borderDisabled,
                backgroundColor: theme.colors.pp.bgDisabled
              }
            }
          }
        },
        thumb: {
          height: 10,
          width: 10
        }
      }
    },
    HvSelect: {
      classes: {
        root: {
          "&& .HvButton-secondarySubtle": {
            borderColor: inputColors.border,
            backgroundColor: inputColors.bg
          },
          "&& .HvDropdownButton-openUp": {
            borderRadius: `0 0 ${theme.radii.base} ${theme.radii.base}`
          },
          "&& .HvDropdownButton-openDown": {
            borderRadius: `${theme.radii.base} ${theme.radii.base} 0 0`
          }
        },
        select: {
          borderRadius: theme.radii.base
        },
        panel: {
          borderColor: inputColors.border
        }
      }
    },
    HvSuggestions: {
      classes: {
        list: {
          borderColor: inputColors.border
        }
      }
    },
    HvTag: {
      classes: {
        root: {
          borderRadius: theme.radii.full,
          padding: theme.spacing("2px", 0)
        },
        label: {
          paddingLeft: 8,
          paddingRight: 8
        },
        icon: {
          marginLeft: theme.space.xs
        },
        deleteIcon: {
          borderRadius: "inherit",
          paddingRight: 4
        },
        selected: {}
      }
    },
    HvInlineEditor: {
      classes: {
        button: {
          borderRadius: 2,
          "&:focus": {
            borderColor: theme.colors.secondary
          }
        },
        inputRoot: {
          borderRadius: 2
        }
      }
    },
    HvButton: {
      classes: {
        root: {
          "--radius": theme.radii.full,
          ":where(:not(.HvButton-disabled,.HvButton-contained))": {
            "&[data-color=positive]": { color: theme.colors.pp.success },
            "&[data-color=warning]": { color: theme.colors.pp.warning },
            "&[data-color=negative]": { color: theme.colors.pp.error },
            ":hover": { backgroundColor: theme.colors.pp.primaryDimmed },
            ":active": { backgroundColor: theme.colors.pp.primarySubtle }
          }
        },
        contained: {
          ":where(:not(.HvButton-disabled))": {
            color: "#FFFFFF",
            "&[data-color=primary]": {
              backgroundColor: colorsPalette.blue[600],
              ":hover": { backgroundColor: colorsPalette.blue[700] },
              ":active": { backgroundColor: colorsPalette.blue[800] }
            },
            "&[data-color=positive]": {
              ":hover": { backgroundColor: theme.colors.pp.successAction },
              ":active": { backgroundColor: theme.colors.pp.successStrong }
            },
            "&[data-color=warning]": {
              ":hover": { backgroundColor: theme.colors.pp.warningAction },
              ":active": { backgroundColor: theme.colors.pp.warningStrong }
            },
            "&[data-color=negative]": {
              ":hover": { backgroundColor: theme.colors.pp.errorAction },
              ":active": { backgroundColor: theme.colors.pp.errorStrong }
            }
          }
        },
        subtle: {
          borderColor: "color-mix(in srgb, currentcolor, transparent 60%)",
          ":where(:not(.HvButton-disabled))": {
            "&[data-color=primary]": {
              borderColor: buttonColors.primary.subtleBorder,
              backgroundColor: buttonColors.primary.subtleBg
            },
            "&[data-color=secondary]": {
              borderColor: buttonColors.secondary.subtleBorder,
              backgroundColor: buttonColors.secondary.subtleBg
            },
            ":hover": {
              backgroundColor: theme.colors.pp.primaryDimmed
            },
            ":active": {
              borderColor: "transparent",
              backgroundColor: theme.colors.pp.primarySubtle
            },
            "&[data-color=positive]": {
              borderColor: buttonColors.success.subtleBorder,
              backgroundColor: theme.colors.pp.successDimmed,
              ":hover": { backgroundColor: buttonColors.success.subtleBg },
              ":active": { backgroundColor: buttonColors.success.subtleBorder }
            },
            "&[data-color=warning]": {
              borderColor: buttonColors.warning.subtleBorder,
              backgroundColor: theme.colors.pp.warningDimmed,
              ":hover": { backgroundColor: buttonColors.warning.subtleBg },
              ":active": { backgroundColor: buttonColors.warning.subtleBorder }
            },
            "&[data-color=negative]": {
              borderColor: buttonColors.error.subtleBorder,
              backgroundColor: theme.colors.pp.errorDimmed,
              ":hover": { backgroundColor: buttonColors.error.subtleBg },
              ":active": { backgroundColor: buttonColors.error.subtleBorder }
            }
          }
        },
        ghost: {
          ":where(:not(.HvButton-disabled))": {
            "&[data-color=positive]": {
              ":hover": { backgroundColor: theme.colors.pp.successDimmed },
              ":active": { backgroundColor: buttonColors.success.subtleBg }
            },
            "&[data-color=warning]": {
              ":hover": { backgroundColor: theme.colors.pp.warningDimmed },
              ":active": { backgroundColor: buttonColors.warning.subtleBg }
            },
            "&[data-color=negative]": {
              ":hover": { backgroundColor: theme.colors.pp.errorDimmed },
              ":active": { backgroundColor: buttonColors.error.subtleBg }
            }
          }
        },
        semantic: {},
        disabled: {
          color: theme.colors.pp.textDisabled,
          ":not(.HvButton-ghost)": {
            borderColor: "transparent",
            backgroundColor: theme.colors.pp.bgDisabled,
            "&:hover, &:active": {
              backgroundColor: theme.colors.pp.bgDisabled
            }
          }
        }
      }
    },
    HvMultiButton: {
      classes: {
        multiple: {
          borderRadius: theme.radii.full,
          borderColor: buttonColors.secondary.subtleBorder,
          "& .HvMultiButton-button": {
            borderColor: "inherit",
            ...theme.typography.body,
            "&.HvMultiButton-firstButton": {
              borderRadius: `${theme.radii.full} 0 0 ${theme.radii.full}`
            },
            "&.HvMultiButton-lastButton": {
              borderRadius: `0 ${theme.radii.full} ${theme.radii.full} 0`
            },
            "&.HvMultiButton-selected": {
              borderColor: theme.colors.primary,
              color: theme.colors.primary,
              backgroundColor: theme.colors.pp.primaryDimmed
            },
            "&:not(.HvMultiButton-firstButton, .HvMultiButton-lastButton)": {
              borderRadius: 0
            }
          }
        },
        vertical: {
          borderColor: buttonColors.secondary.subtleBorder,
          "& .HvMultiButton-button": {
            borderColor: "inherit",
            "&.HvMultiButton-firstButton": {
              borderRadius: "16px 16px 0 0"
            },
            "&.HvMultiButton-lastButton": {
              borderRadius: "0 0 16px 16px"
            },
            "&.HvMultiButton-selected": {
              borderColor: theme.colors.primary,
              color: theme.colors.primary,
              backgroundColor: theme.colors.pp.primaryDimmed
            }
          }
        },
        splitGroup: {},
        splitGroupDisabled: {}
      }
    },
    HvDropdownButton: {
      classes: {
        openUp: {
          borderRadius: "0px 0px calc(var(--HvButton-height) / 2) calc(var(--HvButton-height) / 2)"
        },
        openDown: {
          borderRadius: "calc(var(--HvButton-height) / 2) calc(var(--HvButton-height) / 2) 0px 0px"
        },
        disabled: {
          backgroundColor: theme.colors.pp.bgDisabled,
          borderColor: theme.colors.pp.bgDisabled,
          "&.HvButton-subtle": {
            backgroundColor: theme.colors.pp.bgDisabled,
            borderColor: theme.colors.pp.bgDisabled,
            "&:hover": {
              backgroundColor: theme.colors.pp.bgDisabled
            }
          },
          "&.HvButton-ghost": {
            backgroundColor: theme.colors.pp.bgDisabled,
            borderColor: theme.colors.pp.bgDisabled,
            "&:hover": {
              backgroundColor: theme.colors.pp.bgDisabled
            }
          }
        }
      }
    },
    HvHeader: {
      classes: {
        root: {
          borderBottom: `1px solid ${theme.colors.atmo3}`,
          boxShadow: "none"
        }
      }
    },
    HvHeaderBrand: {
      classes: {
        separator: {
          backgroundColor: theme.colors.atmo4,
          margin: theme.spacing(0, "md"),
          height: 32
        }
      }
    },
    HvHeaderMenuBarBar: {
      classes: {
        active: {
          boxShadow: "none",
          borderBottom: `1px solid ${theme.colors.atmo3}`
        },
        list: {
          "& li:hover > .HvHeader-MenuBar-hidden": {
            boxShadow: "none",
            borderBottom: `1px solid ${theme.colors.atmo3}`
          },
          "& li:focus-within > .HvHeader-MenuBar-hidden": {
            boxShadow: "none",
            borderBottom: `1px solid ${theme.colors.atmo3}`
          }
        }
      }
    },
    HvVerticalNavigation: {
      classes: {
        root: {
          color: theme.colors.base_light,
          backgroundColor: colorsPalette.slate[900],
          borderRight: `1px solid ${colorsPalette.slate[500]}`,
          "& > :not(nav:first-of-type)": {
            borderTop: `1px solid ${colorsPalette.slate[500]}`
          },
          "& > :only-child": {
            padding: theme.space.sm,
            "& .HvVerticalNavigationSlider-listContainer": { padding: 0 }
          }
        },
        slider: {
          "& > div:first-of-type": {
            borderBottom: `1px solid ${colorsPalette.slate[500]}`
          }
        }
      }
    },
    HvVerticalNavigationAction: {
      classes: {
        action: {
          borderRadius: "8px",
          "&:hover, &:focus": {
            backgroundColor: colorsPalette.slate[700]
          }
        }
      }
    },
    HvVerticalNavigationSlider: {
      classes: {
        root: {
          minHeight: "32px",
          borderLeft: "unset",
          borderRadius: "8px",
          "&.HvIsFocused": {
            backgroundColor: colorsPalette.slate[700]
          },
          "&.HvListItem-interactive:not(.HvListItem-disabled):not(.HvListItem-selected):hover": {
            backgroundColor: colorsPalette.slate[700]
          }
        },
        listItemDisabled: {
          color: colorsPalette.neutral[500],
          backgroundColor: colorsPalette.neutral[800],
          "& .HvListItem-startAdornment": { backgroundColor: "transparent" },
          "& .HvListItem-endAdornment": { backgroundColor: "transparent" }
        },
        listItemSelected: {
          background: colorsPalette.blue[950],
          borderLeft: "unset"
        },
        listItemFocus: {
          background: colorsPalette.slate[700]
        },
        listContainer: {
          padding: theme.space.sm
        }
      }
    },
    HvVerticalNavigationTreeViewItem: {
      classes: {
        content: {
          borderLeft: "unset",
          borderRadius: "8px",
          ".HvVerticalNavigationTreeViewItem-selected>&": {
            background: colorsPalette.blue[950],
            borderLeft: "unset"
          },
          ":not(.HvVerticalNavigationTreeViewItem-disabled>&):not(.HvVerticalNavigationTreeViewItem-selected>&)": {
            "&:hover, &:focus-visible, &.focus-visible": {
              background: colorsPalette.slate[700]
            }
          },
          ".HvVerticalNavigationTreeViewItem-focused>&": {
            background: colorsPalette.slate[700]
          }
        },
        disabled: {
          "& .HvVerticalNavigationTreeViewItem-label": {
            color: colorsPalette.neutral[500]
          },
          "& .HvVerticalNavigationTreeViewItem-content": {
            background: colorsPalette.neutral[800]
          }
        },
        icon: {
          "& .HvAvatar-root": {
            borderRadius: "8px"
          }
        }
      }
    },
    HvCard: {
      classes: {
        root: {
          outline: `1px solid ${theme.colors.pp.dividerSubtle}`,
          overflow: "hidden",
          height: "fit-content",
          borderRadius: theme.space.sm,
          "& > :last-child:not(.HvCardMedia-root)": {
            paddingBottom: theme.space.sm
          },
          "& .HvActionBar-root": {
            padding: theme.spacing("xs", "sm"),
            borderTop: "none"
          }
        },
        selectable: {
          ":hover": {
            outline: `1px solid ${theme.colors.primary_20}`,
            backgroundColor: theme.colors.pp.primaryDimmed
          }
        },
        semanticContainer: {
          display: "none"
        },
        selected: {
          outline: `1px solid ${theme.colors.pp.primaryAction}`
        }
      }
    },
    HvCardHeader: {
      classes: {
        root: {
          flexDirection: "row-reverse",
          padding: `${theme.space.xs} ${theme.space.sm}`
        },
        action: {
          paddingLeft: 0
        }
      }
    },
    HvCardContent: {
      classes: {
        content: {
          padding: `${theme.space.xs} ${theme.space.sm}`,
          "&:last-child": {
            paddingBottom: theme.space.xs
          }
        }
      }
    },
    HvFooter: {
      name: "Pentaho+"
    },
    HvTabs: {
      classes: {
        floating: {
          "& .HvTab-root": {
            marginTop: 0,
            zIndex: 1,
            "&:is(.HvTab-selected)": {
              borderColor: "transparent",
              backgroundColor: "transparent"
            },
            "&:hover": {
              borderRadius: theme.radii.full
            },
            "::after": {
              display: "none"
            }
          },
          "& .HvTabs-indicator": {
            height: "100%",
            backgroundColor: theme.colors.pp.bgSurface,
            border: `1px solid ${theme.colors.primary}`,
            borderRadius: theme.radii.full
          },
          "& .HvTabs-flexContainer": {
            display: "inline-flex",
            backgroundColor: theme.colors.pp.bgActive,
            borderRadius: theme.radii.full,
            marginLeft: 0
          }
        }
      }
    },
    HvBaseInput: {
      classes: {
        root: {
          borderColor: inputColors.border,
          backgroundColor: inputColors.bg
        }
      }
    },
    HvBaseDropdown: {
      classes: {
        header: {
          borderColor: inputColors.border,
          backgroundColor: inputColors.bg
        },
        headerOpen: {
          borderColor: inputColors.border,
          "&:hover": {
            borderColor: inputColors.border
          }
        },
        panel: {
          borderColor: inputColors.border
        }
      }
    },
    HvDropDownMenu: {
      classes: {
        iconSelected: {
          "&[data-color=secondary]": {
            borderColor: inputColors.border
          }
        }
      }
    }
  },
  header: {
    height: "64px",
    secondLevelHeight: "56px"
  },
  bulkActions: {
    actionButtonVariant: "primaryGhost"
  },
  table: {
    rowStripedBackgroundColorEven: theme.colors.atmo1,
    // TODO - remove in v6
    rowStripedBackgroundColorOdd: "transparent",
    // TODO - remove in v6
    rowExpandBackgroundColor: theme.colors.atmo2,
    rowSortedColor: theme.colors.primary,
    // TODO - remove in v6
    rowSortedColorAlpha: "0.1"
    // TODO - remove in v6
  },
  stepNavigation: {
    separatorMargin: "4px",
    defaultSeparatorHeight: 1,
    simpleSeparatorHeight: 1
  },
  filterGroup: {
    applyButtonVariant: "primary",
    cancelButtonVariant: "secondarySubtle"
  },
  scrollTo: {
    dotSelectedSize: 6,
    backgroundColorOpacity: 0.9
    // TODO - remove in v6
  },
  colorPicker: {
    hueDirection: "horizontal"
  },
  snackbar: {
    actionButtonVariant: "secondarySubtle"
  }
}));
exports.default = pentahoPlus;
