// @hitachivantara/uikit-react-core/dist/cjs/VerticalNavigation/Actions/Action.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const setId = require("../../utils/setId.cjs");
const VerticalNavigationContext = require("../VerticalNavigationContext.cjs");
const Action_styles = require("./Action.styles.cjs");
const Button = require("../../Button/Button.cjs");
const HvVerticalNavigationAction = (props) => {
  const {
    className,
    classes: classesProp,
    id,
    label = "",
    icon,
    ...others
  } = uikitReactUtils.useDefaultProps("HvVerticalNavigationAction", props);
  const { isOpen } = React.useContext(VerticalNavigationContext.VerticalNavigationContext);
  const { classes, cx } = Action_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsxs(
    Button.HvButton,
    {
      id: setId.setId(id, "button"),
      variant: "secondaryGhost",
      icon: !isOpen,
      className: cx(
        classes.action,
        {
          [classes.noIcon]: !icon,
          [classes.minimized]: !isOpen
        },
        className
      ),
      ...!isOpen && { "aria-label": label },
      ...others,
      children: [
        icon,
        isOpen && label
      ]
    }
  );
};
exports.actionClasses = Action_styles.staticClasses;
exports.HvVerticalNavigationAction = HvVerticalNavigationAction;
