// @hitachivantara/uikit-react-core/dist/cjs/BaseRadio/BaseRadio.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvBaseRadio", {
  root: {
    padding: 0,
    width: 32,
    minWidth: 32,
    height: 32,
    cursor: "pointer",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
      borderRadius: uikitStyles.theme.radii.base
    },
    "& svg": {
      width: 16,
      height: 16,
      borderRadius: uikitStyles.theme.radii.circle,
      borderWidth: 1,
      borderColor: uikitStyles.theme.colors.secondary,
      backgroundColor: uikitStyles.theme.colors.atmo1
    },
    borderRadius: 0
  },
  disabled: {
    cursor: "not-allowed",
    pointerEvents: "initial",
    "& svg": {
      borderColor: uikitStyles.theme.colors.secondary_60,
      backgroundColor: uikitStyles.theme.colors.atmo3
    }
  },
  focusVisible: {
    "& svg": {
      borderRadius: "8px",
      ...focusUtils.outlineStyles
    }
  },
  icon: {},
  checked: {
    "& svg": {
      borderColor: uikitStyles.theme.colors.secondary,
      backgroundColor: uikitStyles.theme.colors.secondary,
      color: uikitStyles.theme.colors.atmo2
    },
    "&$semantic": {
      "& svg": {
        borderColor: uikitStyles.theme.colors.base_dark,
        backgroundColor: uikitStyles.theme.colors.base_light,
        color: uikitStyles.theme.colors.base_dark
      }
    },
    "&$disabled": {
      "& svg": {
        borderColor: uikitStyles.theme.colors.secondary_60,
        backgroundColor: uikitStyles.theme.colors.secondary_60,
        color: uikitStyles.theme.colors.atmo3
      }
    }
  },
  semantic: {
    "& svg": {
      borderColor: uikitStyles.theme.colors.base_dark,
      backgroundColor: uikitStyles.theme.colors.base_light
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
