// @hitachivantara/uikit-react-core/dist/cjs/BaseRadio/BaseRadio.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const MuiRadio = require("@mui/material/Radio");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const BaseRadio_styles = require("./BaseRadio.styles.cjs");
const icons = require("./icons.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiRadio__default = /* @__PURE__ */ _interopDefault(MuiRadio);
const getSelectorIcons = () => {
  return {
    radio: /* @__PURE__ */ jsxRuntime.jsx(icons.Unselected, {}),
    radioChecked: /* @__PURE__ */ jsxRuntime.jsx(icons.Selected, {})
  };
};
const HvBaseRadio = React.forwardRef(
  function HvBaseRadio2(props, ref) {
    const {
      classes: classesProp,
      className,
      id,
      name,
      value = "on",
      required = false,
      readOnly = false,
      disabled = false,
      checked,
      defaultChecked,
      onChange,
      semantic = false,
      inputProps,
      onFocusVisible,
      onBlur,
      icon,
      checkedIcon,
      ...others
    } = uikitReactUtils.useDefaultProps("HvBaseRadio", props);
    const { classes, cx } = BaseRadio_styles.useClasses(classesProp);
    const [focusVisible, setFocusVisible] = React.useState(false);
    const onFocusVisibleCallback = React.useCallback(
      (evt) => {
        setFocusVisible(true);
        onFocusVisible?.(evt);
      },
      [onFocusVisible]
    );
    const onBlurCallback = React.useCallback(
      (evt) => {
        setFocusVisible(false);
        onBlur?.(evt);
      },
      [onBlur]
    );
    const icons2 = getSelectorIcons();
    const onLocalChange = React.useCallback(
      (evt) => {
        if (readOnly) {
          return;
        }
        onChange?.(evt, evt.target.checked, value);
      },
      [onChange, readOnly, value]
    );
    return /* @__PURE__ */ jsxRuntime.jsx(
      MuiRadio__default.default,
      {
        ref,
        id,
        name,
        className: cx(
          classes.root,
          {
            [classes.disabled]: disabled,
            [classes.focusVisible]: focusVisible,
            [classes.checked]: checked,
            [classes.semantic]: semantic
          },
          className
        ),
        icon: icon || icons2.radio,
        checkedIcon: checkedIcon || icons2.radioChecked,
        color: "default",
        disabled,
        required,
        readOnly,
        onChange: onLocalChange,
        value,
        checked,
        defaultChecked,
        inputProps,
        onFocusVisible: onFocusVisibleCallback,
        onBlur: onBlurCallback,
        ...others
      }
    );
  }
);
exports.baseRadioClasses = BaseRadio_styles.staticClasses;
exports.HvBaseRadio = HvBaseRadio;
exports.getSelectorIcons = getSelectorIcons;
