// @hitachivantara/uikit-react-core/dist/cjs/Banner/BannerContent/BannerContent.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvBannerContent", {
  root: {
    minWidth: "100%",
    position: "relative",
    display: "flex",
    flexWrap: "nowrap",
    boxShadow: "none",
    borderRadius: uikitStyles.theme.radii.none
  },
  message: {
    display: "flex",
    alignItems: "center",
    padding: uikitStyles.theme.spacing("xs", 0),
    paddingLeft: uikitStyles.theme.space.sm,
    ...uikitStyles.theme.typography.body,
    color: uikitStyles.theme.colors.base_dark
  },
  action: {
    padding: uikitStyles.theme.spacing("xs"),
    marginRight: 0,
    flex: "0 0 auto",
    placeSelf: "stretch"
  },
  baseVariant: {
    padding: 0
  },
  success: {
    backgroundColor: uikitStyles.theme.colors.positive_20
  },
  warning: {
    backgroundColor: uikitStyles.theme.colors.warning_20
  },
  error: {
    backgroundColor: uikitStyles.theme.colors.negative_20
  },
  default: {
    backgroundColor: uikitStyles.theme.colors.neutral_20
  },
  outContainer: {
    width: "100%",
    position: "relative"
  },
  messageContainer: {
    wordBreak: "break-word",
    maxWidth: 700,
    overflow: "hidden",
    marginRight: 10
  },
  iconContainer: {
    marginRight: uikitStyles.theme.space.xs,
    marginLeft: -uikitStyles.theme.space.xs
  },
  messageActions: {
    flex: "0 0 auto"
  },
  actionContainer: {
    display: "flex",
    flexDirection: "column",
    height: "100%",
    justifyContent: "space-between"
  },
  actionsInnerContainer: {
    flexDirection: "row",
    marginTop: "8px"
    // avoid overlap with close button outline focus ring
  },
  closeAction: {
    alignSelf: "flex-end"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
