// @hitachivantara/uikit-react-core/dist/cjs/Footer/Footer.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const styles = require("@mui/material/styles");
const useMediaQuery = require("@mui/material/useMediaQuery");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const Footer_styles = require("./Footer.styles.cjs");
const Typography = require("../Typography/Typography.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const useMediaQuery__default = /* @__PURE__ */ _interopDefault(useMediaQuery);
const HvFooter = React.forwardRef(function HvFooter2(props, ref) {
  const {
    name = "Hitachi Vantara",
    copyright = `© Hitachi Vantara LLC ${(/* @__PURE__ */ new Date()).getFullYear()}. All Rights Reserved.`,
    links,
    classes: classesProp,
    className,
    ...others
  } = uikitReactUtils.useDefaultProps("HvFooter", props);
  const muiTheme = styles.useTheme();
  const { classes, cx } = Footer_styles.useClasses(classesProp);
  const isSmDown = useMediaQuery__default.default(muiTheme.breakpoints.down("sm"));
  return /* @__PURE__ */ jsxRuntime.jsxs(
    "footer",
    {
      ref,
      className: cx(classes.root, { [classes.small]: isSmDown }, className),
      ...others,
      children: [
        /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { variant: "label", className: classes.name, children: name }),
        /* @__PURE__ */ jsxRuntime.jsxs("div", { className: classes.rightContainer, children: [
          /* @__PURE__ */ jsxRuntime.jsx(Typography.HvTypography, { className: classes.copyright, children: copyright }),
          links && /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.separator }),
          links
        ] })
      ]
    }
  );
});
exports.footerClasses = Footer_styles.staticClasses;
exports.HvFooter = HvFooter;
