// @hitachivantara/uikit-react-core/dist/cjs/ColorPicker/Picker/Picker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const name = "HvColorPicker-Picker";
const { staticClasses, useClasses } = uikitReactUtils.createClasses(name, {
  pickers: {
    display: "flex",
    flexDirection: "column"
  },
  saturation: {
    width: "232px",
    height: "140px",
    marginRight: "0px",
    position: "relative",
    overflow: "visible",
    "& > div": {
      borderRadius: uikitStyles.theme.radii.base,
      "& .saturation-white": {
        borderRadius: uikitStyles.theme.radii.base,
        "& .saturation-black": {
          borderRadius: uikitStyles.theme.radii.base
        }
      }
    }
  },
  saturationPointer: {
    width: "8px",
    height: "8px",
    boxShadow: `0 0 0 2px #fff, inset 0 0 1px 1px rgba(0,0,0,.3),
            0 0 1px 2px rgba(0,0,0,.4)`,
    borderRadius: "50%",
    transform: "translate(-3px, -3px)"
  },
  hue: {
    height: "8px",
    width: "232px",
    position: "relative",
    overflow: "visible",
    marginTop: "18px",
    "& .hue-horizontal": {
      borderRadius: `calc(2*${uikitStyles.theme.radii.base})`
    }
  },
  hueSlider: {
    width: "12px",
    height: "12px",
    background: "transparent",
    boxShadow: "0 0 2px rgb(0 0 0 / 60%)",
    marginLeft: "0px",
    border: "2px solid #fff",
    borderRadius: uikitStyles.theme.radii.circle,
    "&.vertical": {
      transform: "translate(0px, -2px)"
    },
    "&.horizontal": {
      transform: "translate(-6px, -2px)"
    }
  },
  title: {
    fontWeight: uikitStyles.theme.fontWeights.semibold,
    marginBottom: 8
  },
  fields: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
