// @hitachivantara/uikit-react-core/dist/esm/ColorPicker/SavedColors/SavedColors.js
import { jsxs, jsx } from "react/jsx-runtime";
import { Swatch } from "react-color/lib/components/common";
import { Add, CloseXS } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./SavedColors.styles.js";
import { staticClasses } from "./SavedColors.styles.js";
import { HvButton } from "../../Button/Button.js";
const SavedColors = (props) => {
  const {
    colors,
    onClickColor,
    onAddColor,
    onRemoveColor,
    deleteButtonAriaLabel,
    addButtonAriaLabel,
    classes: classesProp
  } = useDefaultProps("HvColorPickerSavedColors", props);
  const { classes } = useClasses(classesProp);
  const handleClick = (hex) => {
    onClickColor({
      hex,
      source: "hex"
    });
  };
  return /* @__PURE__ */ jsxs("div", { className: classes.root, children: [
    /* @__PURE__ */ jsx(
      HvButton,
      {
        className: classes.addButton,
        variant: "secondarySubtle",
        icon: true,
        onClick: onAddColor,
        "aria-label": addButtonAriaLabel,
        children: /* @__PURE__ */ jsx(Add, { "aria-hidden": true })
      }
    ),
    colors.map((color, index) => /* @__PURE__ */ jsxs(
      "div",
      {
        className: classes.swatchRoot,
        children: [
          /* @__PURE__ */ jsx("div", { className: classes.swatchWrap, children: /* @__PURE__ */ jsx(
            Swatch,
            {
              color,
              onClick: handleClick,
              focusStyle: {
                boxShadow: `inset 0 0 0 1px rgba(0,0,0,.15), 0 0 4px ${color}`
              }
            }
          ) }),
          /* @__PURE__ */ jsx("div", { className: classes.removeButtonRoot, children: /* @__PURE__ */ jsx(
            HvButton,
            {
              className: classes.removeButton,
              variant: "secondarySubtle",
              onClick: () => onRemoveColor(color, index),
              "aria-label": deleteButtonAriaLabel,
              children: /* @__PURE__ */ jsx(CloseXS, { "aria-hidden": true, iconSize: "XS" })
            }
          ) })
        ]
      },
      `saved-color-${color}-${index}`
    ))
  ] });
};
export {
  SavedColors,
  staticClasses as colorPickerSavedColorsClasses
};
