// @hitachivantara/uikit-react-core/dist/cjs/FilterGroup/FilterContent/FilterContent.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const name = "HvFilterGroupContent";
const { staticClasses, useClasses } = uikitReactUtils.createClasses(name, {
  dropdown: {},
  panel: {
    maxHeight: 500,
    minHeight: 370
  },
  header: {
    display: "flex",
    justifyContent: "space-between",
    height: 32
  },
  baseDropdownSelection: {
    padding: uikitStyles.theme.spacing("0px", "30px", "0px", "0px")
  },
  root: {
    width: 640,
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)"
  },
  rightSidePanel: {
    display: "inline-block",
    width: "50%",
    height: "100%",
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)",
    verticalAlign: "top",
    overflow: "visible",
    boxShadow: "none",
    borderLeft: `1px solid ${uikitStyles.theme.colors.atmo3}`
  },
  leftSidePanel: {
    display: "inline-block",
    width: "50%",
    height: "100%",
    verticalAlign: "top",
    maxHeight: "calc(500px - 75px)",
    minHeight: "calc(370px - 75px)"
  },
  actionBar: {},
  space: {
    flex: 1
  },
  applyButton: {
    marginRight: "8px"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
