// @hitachivantara/uikit-react-core/dist/cjs/Snackbar/Snackbar.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const Snackbar = require("@mui/material/Snackbar");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvSnackbar", {
  root: {},
  anchorOriginTopRight: {
    [`&.${Snackbar.snackbarClasses.anchorOriginTopRight}`]: {
      top: uikitStyles.theme.space.xs,
      right: uikitStyles.theme.space.xs
    }
  },
  anchorOriginTopLeft: {
    [`&.${Snackbar.snackbarClasses.anchorOriginTopLeft}`]: {
      top: uikitStyles.theme.space.xs,
      left: uikitStyles.theme.space.xs
    }
  },
  anchorOriginTopCenter: {
    [`&.${Snackbar.snackbarClasses.anchorOriginTopCenter}`]: {
      top: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomCenter: {
    [`&.${Snackbar.snackbarClasses.anchorOriginBottomCenter}`]: {
      bottom: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomLeft: {
    [`&.${Snackbar.snackbarClasses.anchorOriginBottomLeft}`]: {
      bottom: uikitStyles.theme.space.xs,
      left: uikitStyles.theme.space.xs
    }
  },
  anchorOriginBottomRight: {
    [`&.${Snackbar.snackbarClasses.anchorOriginBottomRight}`]: {
      bottom: uikitStyles.theme.space.xs,
      right: uikitStyles.theme.space.xs
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
