// @hitachivantara/uikit-react-core/dist/esm/Footer/Footer.js
import { jsxs, jsx } from "react/jsx-runtime";
import { forwardRef } from "react";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./Footer.styles.js";
import { staticClasses } from "./Footer.styles.js";
import { HvTypography } from "../Typography/Typography.js";
const HvFooter = forwardRef(function HvFooter2(props, ref) {
  const {
    name = "Hitachi Vantara",
    copyright = `© Hitachi Vantara LLC ${(/* @__PURE__ */ new Date()).getFullYear()}. All Rights Reserved.`,
    links,
    classes: classesProp,
    className,
    ...others
  } = useDefaultProps("HvFooter", props);
  const muiTheme = useTheme();
  const { classes, cx } = useClasses(classesProp);
  const isSmDown = useMediaQuery(muiTheme.breakpoints.down("sm"));
  return /* @__PURE__ */ jsxs(
    "footer",
    {
      ref,
      className: cx(classes.root, { [classes.small]: isSmDown }, className),
      ...others,
      children: [
        /* @__PURE__ */ jsx(HvTypography, { variant: "label", className: classes.name, children: name }),
        /* @__PURE__ */ jsxs("div", { className: classes.rightContainer, children: [
          /* @__PURE__ */ jsx(HvTypography, { className: classes.copyright, children: copyright }),
          links && /* @__PURE__ */ jsx("div", { className: classes.separator }),
          links
        ] })
      ]
    }
  );
});
export {
  HvFooter,
  staticClasses as footerClasses
};
