// @hitachivantara/uikit-react-core/dist/esm/hocs/withTooltip.js
import { jsx } from "react/jsx-runtime";
import { useState } from "react";
import { HvTooltip } from "../Tooltip/Tooltip.js";
const styles = {
  truncate: {
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap"
  }
};
const withTooltip = (Component, label = "", placement, hideTooltip, tooltipProps, tooltipContainerProps) => (
  // eslint-disable-next-line react/display-name
  (props) => {
    const [isHoverDisabled, setIsHoverDisabled] = useState(
      false
    );
    const [open, setOpen] = useState(false);
    const handleMouseEnter = (event) => {
      const isHidden = hideTooltip?.(event);
      setIsHoverDisabled(isHidden);
      setOpen(!isHidden);
    };
    const handleMouseLeave = () => {
      setIsHoverDisabled(false);
      setOpen(false);
    };
    return /* @__PURE__ */ jsx(
      HvTooltip,
      {
        style: { ...styles.truncate },
        disableHoverListener: isHoverDisabled,
        disableFocusListener: true,
        disableTouchListener: true,
        title: label,
        open,
        placement,
        ...tooltipProps,
        children: /* @__PURE__ */ jsx(
          "div",
          {
            ...tooltipContainerProps,
            onMouseEnter: handleMouseEnter,
            onMouseLeave: handleMouseLeave,
            children: /* @__PURE__ */ jsx(Component, { ...props })
          }
        )
      }
    );
  }
);
export {
  withTooltip
};
