// @hitachivantara/uikit-react-core/dist/esm/TableSection/TableSection.styles.js
import { createClasses } from "@hitachivantara/uikit-react-utils";
import { theme } from "@hitachivantara/uikit-styles";
import { staticClasses as staticClasses$1 } from "../Pagination/Pagination.styles.js";
import "../Pagination/Pagination.js";
import { staticClasses as staticClasses$2 } from "../BulkActions/BulkActions.styles.js";
import "../BulkActions/BulkActions.js";
import { staticClasses as staticClasses$3 } from "../Table/TableCell/TableCell.styles.js";
import "../Table/TableCell/TableCell.js";
import { staticClasses as staticClasses$4 } from "../Table/TableRow/TableRow.styles.js";
import "../Table/TableRow/TableRow.js";
import { staticClasses as staticClasses$5 } from "../Table/TableHeader/TableHeader.styles.js";
import "../Table/TableHeader/TableHeader.js";
import { staticClasses as staticClasses$6 } from "../Table/TableContainer/TableContainer.styles.js";
import "../Table/TableContainer/TableContainer.js";
const { staticClasses, useClasses } = createClasses("HvTableSection", {
  root: {},
  header: {
    // Only apply the border to divide the header and content when both are displayed
    "+ div": {
      borderTopLeftRadius: 0,
      borderTopRightRadius: 0
    }
  },
  actions: {},
  content: {
    marginTop: 0,
    padding: 0,
    // Apply border radius to the first child if there's not an header
    "&:first-of-type": {
      "& > :first-of-type": {
        borderTopLeftRadius: "inherit",
        borderTopRightRadius: "inherit"
      }
    },
    "& > :last-child": {
      borderBottomLeftRadius: "inherit",
      borderBottomRightRadius: "inherit"
    },
    [`& .${staticClasses$6.root}`]: {
      paddingBottom: 0,
      borderTopLeftRadius: "inherit",
      borderTopRightRadius: "inherit"
    },
    [`& .${staticClasses$5.head}`]: {
      backgroundColor: theme.colors.atmo2,
      borderBottomColor: theme.colors.atmo3,
      [`&.${staticClasses$5.variantCheckbox}`]: {
        borderRight: "none"
      },
      [`&.${staticClasses$5.variantActions}`]: {
        borderLeft: "none"
      }
    },
    // Remove border for the last table row
    [`& .${staticClasses$4.root}`]: {
      "&:last-child": {
        [`& .${staticClasses$3.root}`]: { borderBottom: "none" }
      }
    },
    [`& .${staticClasses$3.root}`]: {
      borderBottomColor: theme.colors.atmo3
    },
    [`& .${staticClasses$3.variantCheckbox}`]: {
      borderRight: "none"
    },
    [`& .${staticClasses$3.variantActions}`]: {
      borderLeft: "none"
    },
    [`& .${staticClasses$2.root}`]: {
      marginBottom: 0,
      border: "none",
      borderBottom: `1px solid ${theme.colors.atmo3}`,
      padding: theme.spacing("xs", "sm")
    },
    [`& .${staticClasses$1.root}`]: {
      margin: 0,
      backgroundColor: theme.colors.atmo2,
      padding: theme.space.xs,
      borderTop: `1px solid ${theme.colors.atmo3}`,
      borderBottomLeftRadius: "inherit",
      borderBottomRightRadius: "inherit"
    },
    [`& .${staticClasses$1.pageSizeOptions}`]: {
      left: theme.space.sm
    },
    [`& .${staticClasses$1.pageSizeHeader}`]: {
      border: "none",
      "&:hover": {
        border: "none"
      }
    }
  },
  hidden: {},
  raisedHeader: {
    "& $content": {
      paddingTop: 0
    }
  },
  hasHeader: {},
  /** @deprecated use `hasHeader` instead */
  spaceTop: {}
});
export {
  staticClasses,
  useClasses
};
