// @hitachivantara/uikit-react-core/dist/cjs/Table/TableHead/TableHead.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const TableContext = require("../TableContext.cjs");
const TableSectionContext = require("../TableSectionContext.cjs");
const TableHead_styles = require("./TableHead.styles.cjs");
const tableSectionContext = {
  type: "head"
};
const defaultComponent = "thead";
const HvTableHead = React.forwardRef(
  function HvTableHead2(props, ref) {
    const {
      classes: classesProp,
      className,
      component,
      stickyHeader,
      ...others
    } = uikitReactUtils.useDefaultProps("HvTableHead", props);
    const { classes, cx } = TableHead_styles.useClasses(classesProp);
    const tableContext = React.useContext(TableContext.default);
    const Component = component || tableContext?.components?.THead || defaultComponent;
    return /* @__PURE__ */ jsxRuntime.jsx(TableSectionContext.TableSectionContext.Provider, { value: tableSectionContext, children: /* @__PURE__ */ jsxRuntime.jsx(
      Component,
      {
        className: cx(
          classes.root,
          { [classes.stickyHeader]: stickyHeader },
          className
        ),
        ref,
        role: Component === defaultComponent ? null : "rowgroup",
        ...others
      }
    ) });
  }
);
exports.tableHeadClasses = TableHead_styles.staticClasses;
exports.HvTableHead = HvTableHead;
