// @hitachivantara/uikit-react-core/dist/cjs/ColorPicker/ColorPicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvColorPicker", {
  root: {},
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {},
  description: {},
  headerColorValue: {
    textTransform: "uppercase",
    minWidth: "8ch",
    fontVariant: "tabular-nums"
  },
  headerColorIcon: {
    width: 16,
    height: 16,
    marginRight: 8,
    flexShrink: 0
  },
  panel: {
    width: "100%",
    minWidth: "266px",
    display: "flex",
    justifyContent: "center",
    padding: "16px",
    backgroundColor: "transparent"
  },
  colorPicker: {
    width: "232px"
  },
  colorPickerIcon: {},
  recommendedColorsRoot: {
    ":not(:only-child)": {
      paddingBottom: "24px"
    }
  },
  dropdownRootIconOnly: {
    width: 32,
    height: 32,
    "& .HvBaseDropdown-selection": {
      padding: 0
    }
  },
  headerColorIconOnly: {
    margin: 8
  },
  pickerFields: { paddingBottom: 20 }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
