// @hitachivantara/uikit-react-core/dist/cjs/InlineEditor/InlineEditor.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvInlineEditor", {
  root: {},
  /** @deprecated unused. use `classes.root::after` instead */
  inputBorderContainer: {},
  input: {},
  inputRoot: {
    height: "100%",
    minHeight: "32px"
  },
  text: {},
  largeText: {},
  textEmpty: {
    color: uikitStyles.theme.colors.secondary_60
  },
  button: {
    padding: uikitStyles.theme.spacing("6px", "8px", "5px", "8px"),
    minHeight: "32px",
    boxSizing: "border-box",
    cursor: "text",
    height: "100%",
    width: "100%",
    maxWidth: "100%",
    justifyContent: "start",
    textAlign: "start",
    alignItems: "center",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    borderColor: "transparent",
    "&:hover, &:focus": {
      borderColor: uikitStyles.theme.colors.primary,
      backgroundColor: uikitStyles.theme.colors.atmo1,
      "& $icon": {
        visibility: "visible"
      }
    },
    "&:active": {
      borderColor: uikitStyles.theme.colors.secondary,
      backgroundColor: "transparent",
      "& $icon": {
        visibility: "visible"
      }
    }
  },
  icon: {
    cursor: "pointer",
    visibility: "hidden",
    alignSelf: "center",
    height: 16
  },
  iconVisible: {
    visibility: "visible"
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
