// @hitachivantara/uikit-react-core/dist/cjs/BaseCheckBox/BaseCheckBox.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvBaseCheckBox", {
  root: {
    padding: 0,
    width: 32,
    minWidth: 32,
    height: 32,
    borderRadius: uikitStyles.theme.radii.base,
    cursor: "pointer",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
    },
    "& svg": {
      width: 16,
      height: 16,
      color: uikitStyles.theme.colors.atmo1,
      borderRadius: uikitStyles.theme.radii.none,
      border: `1px solid ${uikitStyles.theme.colors.secondary}`
    }
  },
  disabled: {
    "&$root": {
      cursor: "not-allowed",
      pointerEvents: "initial",
      "& svg": {
        color: uikitStyles.theme.colors.atmo3,
        borderColor: uikitStyles.theme.colors.secondary_60,
        backgroundColor: uikitStyles.theme.colors.atmo3
      }
    }
  },
  focusVisible: {
    "& svg": {
      ...focusUtils.outlineStyles
    }
  },
  icon: {},
  checked: {
    "& svg": {
      border: `1px solid ${uikitStyles.theme.colors.secondary}`,
      backgroundColor: uikitStyles.theme.colors.secondary,
      color: uikitStyles.theme.colors.atmo1
    },
    "&$disabled": {
      "& svg": {
        color: uikitStyles.theme.colors.secondary_60
      }
    }
  },
  indeterminate: {
    "& svg": {
      color: uikitStyles.theme.colors.secondary
    },
    "&$disabled": {
      "& svg": {
        color: uikitStyles.theme.colors.secondary_60
      }
    }
  },
  semantic: {
    "& svg": {
      border: `1px solid ${uikitStyles.theme.colors.base_dark}`,
      color: uikitStyles.theme.colors.base_light,
      backgroundColor: uikitStyles.theme.colors.base_dark
    },
    "&$indeterminate": {
      "& svg": {
        color: uikitStyles.theme.colors.base_dark,
        backgroundColor: uikitStyles.theme.colors.base_light
      }
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
