// @hitachivantara/uikit-react-core/dist/esm/Card/Header/Header.js
import { jsx } from "react/jsx-runtime";
import MuiCardHeader from "@mui/material/CardHeader";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./Header.styles.js";
import { staticClasses } from "./Header.styles.js";
const HvCardHeader = (props) => {
  const {
    classes: classesProp,
    className,
    title,
    subheader,
    icon,
    onClick,
    ...others
  } = useDefaultProps("HvCardHeader", props);
  const { classes, cx } = useClasses(classesProp);
  return /* @__PURE__ */ jsx(
    MuiCardHeader,
    {
      title,
      subheader,
      action: icon,
      onClick,
      className: cx(classes.root, className),
      classes: {
        title: cx({
          // TODO: review conditional in v6
          [classes.titleShort]: icon,
          [classes.title]: !icon
        }),
        subheader: classes.subheader,
        action: classes.action,
        content: classes.content
      },
      ...others
    }
  );
};
export {
  HvCardHeader,
  staticClasses as cardHeaderClasses
};
