// @hitachivantara/uikit-react-core/dist/cjs/Tag/Tag.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvTag", {
  root: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center",
    cursor: "default",
    color: uikitStyles.theme.colors.base_dark,
    backgroundColor: "var(--bgColor)",
    height: 16,
    borderRadius: 0,
    maxWidth: 180,
    whiteSpace: "nowrap",
    ":hover, :focus": {
      backgroundColor: "var(--bgColor)"
    }
  },
  categorical: {
    borderRadius: 8,
    "& $label": {
      color: uikitStyles.theme.colors.secondary
    }
  },
  disabled: {
    backgroundColor: uikitStyles.theme.colors.atmo3,
    ":hover, :focus": {
      backgroundColor: uikitStyles.theme.colors.atmo3
    },
    "& $label": {
      color: uikitStyles.theme.colors.secondary_60
    }
  },
  label: {
    padding: uikitStyles.theme.spacing(0, "xxs"),
    color: "inherit"
  },
  deleteIcon: {
    width: 16,
    height: 16,
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
    }
  },
  selected: {},
  clickable: {
    cursor: "pointer"
  },
  icon: {
    width: 12,
    height: 12,
    marginLeft: 2
  },
  /** @deprecated use `root` instead */
  chipRoot: {},
  /** @deprecated unused */
  // TODO: redundant - use deleteIcon. remove in v6
  /** @deprecated */
  button: {},
  /** @deprecated */
  tagButton: {},
  // TODO: redundant - use $clickable or :not($disabled). remove in v6
  /** @deprecated */
  focusVisible: {},
  /** @deprecated */
  disabledDeleteIcon: {},
  /** @deprecated */
  categoricalFocus: {},
  /** @deprecated */
  categoricalDisabled: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
