// @hitachivantara/uikit-react-core/dist/cjs/ProgressBar/ProgressBar.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvProgressBar", {
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    // progress
    width: "100%"
  },
  progress: {
    width: "100%"
  },
  progressBar: {
    backgroundColor: uikitStyles.theme.colors.secondary
  },
  progressBarLabel: {
    backgroundColor: "transparent",
    display: "flex",
    justifyContent: "end"
  },
  progressContainer: {
    width: "100%"
  },
  progressDone: {
    backgroundColor: uikitStyles.theme.colors.positive
  },
  progressBarContainer: {
    display: "flex",
    width: "100%",
    height: 4,
    backgroundColor: uikitStyles.theme.colors.atmo4
  },
  progressError: {
    backgroundColor: uikitStyles.theme.colors.negative
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
