// @hitachivantara/uikit-react-core/dist/esm/Skeleton/Skeleton.js
import { jsx } from "react/jsx-runtime";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./Skeleton.styles.js";
import { staticClasses } from "./Skeleton.styles.js";
const HvSkeleton = (props) => {
  const {
    classes: classesProp,
    className,
    children,
    hidden = false,
    variant = "text",
    animation = "pulse",
    backgroundImage,
    width,
    height,
    style,
    ...others
  } = useDefaultProps("HvSkeleton", props);
  const { classes, cx } = useClasses(classesProp);
  if (hidden) return /* @__PURE__ */ jsx("div", { className: classes.content, children });
  return /* @__PURE__ */ jsx(
    "div",
    {
      className: cx(
        classes.root,
        classes[variant],
        animation && classes[animation],
        className
      ),
      style: {
        width,
        height,
        ...backgroundImage && {
          backgroundImage: `url(${backgroundImage})`,
          backgroundSize: "contain",
          backgroundPosition: "center",
          backgroundRepeat: "no-repeat"
        },
        ...style
      },
      ...others,
      children
    }
  );
};
export {
  HvSkeleton,
  staticClasses as skeletonClasses
};
