// @hitachivantara/uikit-react-core/dist/cjs/ScrollToVertical/VerticalScrollListItem/VerticalScrollListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const name = "HvVerticalScrollListItem";
const { staticClasses, useClasses } = uikitReactUtils.createClasses(name, {
  root: {
    padding: "0",
    height: "32px",
    width: "32px",
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  },
  icon: {
    width: "1em",
    height: "1em",
    borderRadius: "50%",
    fontSize: 6,
    color: uikitStyles.theme.colors.secondary,
    display: "inline-block",
    backgroundColor: "currentcolor"
  },
  notSelected: {
    fontSize: 4,
    color: uikitStyles.theme.colors.secondary_60
  },
  // TODO: remove in v6 (use classes.button)
  text: {},
  button: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "16px",
    width: "16px",
    borderRadius: "50%",
    cursor: "pointer",
    "&:hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
      "& $notSelected": {
        fontSize: 6,
        color: uikitStyles.theme.colors.secondary
      }
    },
    "&:focus": {
      outline: "none"
    },
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
