// @hitachivantara/uikit-react-core/dist/cjs/Section/Section.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvSection", {
  root: {
    width: "100%",
    display: "flex",
    flexDirection: "column",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    borderRadius: uikitStyles.theme.radii.round,
    border: `1px solid ${uikitStyles.theme.colors.atmo4}`
  },
  hidden: { height: 0, display: "none" },
  header: {
    display: "flex",
    alignItems: "center",
    borderColor: "inherit",
    position: "relative",
    minHeight: uikitStyles.theme.sizes.sm,
    padding: uikitStyles.theme.space.sm
  },
  content: {
    padding: uikitStyles.theme.space.sm,
    borderRadius: "inherit",
    borderColor: "inherit"
  },
  hasHeader: {
    paddingTop: 0
  },
  /** @deprecated use `hasHeader` instead */
  spaceTop: {},
  actions: {
    display: "flex",
    gap: uikitStyles.theme.space.xs,
    marginLeft: "auto"
  },
  raisedHeader: {
    "& $header": {
      zIndex: 1,
      borderBottomWidth: 1,
      boxShadow: uikitStyles.theme.colors.shadow
    },
    "& $content": {
      paddingTop: uikitStyles.theme.space.sm
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
