// @hitachivantara/uikit-react-core/dist/cjs/Controls/LeftControl/LeftControl.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const setId = require("../../utils/setId.cjs");
const ControlsContext = require("../context/ControlsContext.cjs");
const LeftControl_styles = require("./LeftControl.styles.cjs");
const Input = require("../../Input/Input.cjs");
const HvLeftControl = ({
  id,
  classes: classesProp,
  className,
  children,
  placeholder = "Search",
  onSearch,
  hideSearch = false,
  searchProps,
  ...others
}) => {
  const { classes, cx } = LeftControl_styles.useClasses(classesProp);
  const { onSearch: onSearchHandler } = React.useContext(ControlsContext.HvControlsContext);
  const onChangeFilter = (e, value) => {
    onSearch?.(e, value);
    onSearchHandler?.(value);
  };
  return /* @__PURE__ */ jsxRuntime.jsxs("div", { id, className: cx(classes.root, className), ...others, children: [
    !hideSearch && /* @__PURE__ */ jsxRuntime.jsx(
      Input.HvInput,
      {
        id: setId.setId(id, "search-input"),
        type: "search",
        placeholder,
        onChange: onChangeFilter,
        ...searchProps
      }
    ),
    children
  ] });
};
exports.leftControlClasses = LeftControl_styles.staticClasses;
exports.HvLeftControl = HvLeftControl;
