// @hitachivantara/uikit-react-core/dist/cjs/TimePicker/TimePicker.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvTimePicker", {
  root: {
    position: "relative"
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {},
  description: {},
  placeholder: {
    display: "flex",
    gap: 1
  },
  placeholderDisabled: {
    color: uikitStyles.theme.colors.secondary_60
  },
  dropdownHeader: {},
  dropdownHeaderOpen: {},
  dropdownHeaderInvalid: {
    border: `1px solid ${uikitStyles.theme.colors.negative_120}`,
    "&:hover": {
      border: `1px solid ${uikitStyles.theme.colors.negative_120}`
    }
  },
  dropdownPanel: {},
  icon: {},
  timePopperContainer: {
    zIndex: 10,
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    padding: uikitStyles.theme.spacing("xs", 0),
    userSelect: "none",
    minWidth: "175px"
  },
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
