// @hitachivantara/uikit-react-core/dist/cjs/TagsInput/TagsInput.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvTagsInput", {
  /** @deprecated unused */
  listItemGutters: {},
  /** @deprecated use `chipRoot` */
  listItemRoot: {},
  root: { display: "inline-block", width: "100%" },
  // TODO: consider renaming this
  chipRoot: {
    maxWidth: "none"
  },
  disabled: {
    "& $tagsList": {
      backgroundColor: uikitStyles.theme.colors.atmo2,
      "&,:hover": {
        borderColor: uikitStyles.theme.colors.atmo4
      }
    }
  },
  readOnly: {
    "& $tagsList": {
      backgroundColor: uikitStyles.theme.colors.atmo2,
      "&,:hover": {
        borderColor: uikitStyles.theme.colors.secondary_60
      }
    }
  },
  resizable: { width: "auto", resize: "both", overflow: "auto" },
  invalid: {},
  labelContainer: { float: "left", display: "flex", alignItems: "flex-start" },
  label: {},
  description: { display: "block", float: "left" },
  characterCounter: {
    display: "block",
    float: "right",
    textAlign: "right",
    marginBottom: "6px"
  },
  tagsList: {
    display: "flex",
    alignItems: "center",
    alignContent: "flex-start",
    gap: uikitStyles.theme.spacing("xxs", "xs"),
    cursor: "text",
    width: "100%",
    minHeight: 32,
    padding: uikitStyles.theme.spacing("xxs", "xs"),
    overflow: "auto",
    position: "relative",
    flexDirection: "row",
    flexWrap: "wrap",
    backgroundColor: uikitStyles.theme.colors.atmo1,
    borderWidth: 1,
    borderColor: uikitStyles.theme.colors.secondary_80,
    borderRadius: uikitStyles.theme.radii.base,
    "&:hover": {
      borderColor: uikitStyles.theme.colors.primary
    },
    "&:focus-within, &:focus-visible": {
      ...focusUtils.outlineStyles
    },
    "&$singleLine": {
      flexWrap: "nowrap"
    },
    "&$error": {
      borderColor: uikitStyles.theme.colors.negative_120
    },
    "&$invalid": {
      borderColor: uikitStyles.theme.colors.negative_120
    }
  },
  /** @deprecated use `classes.input` instead */
  tagInputContainerRoot: {},
  /** @deprecated use `classes.input` instead */
  tagInputRoot: {},
  input: {
    display: "flex",
    flex: "1 0 auto",
    height: "auto",
    width: 0,
    minWidth: 60,
    border: "none",
    margin: 0,
    padding: 0,
    ...uikitStyles.theme.typography.caption1,
    backgroundColor: "transparent",
    outline: "none",
    boxShadow: "none"
  },
  /** @deprecated unused.  use `:focus` or `:focus-visible` instead */
  tagSelected: {},
  /** @deprecated unused. use `::after` instead */
  tagInputBorderContainer: {},
  /** @deprecated unused. use `:focus` or `:focus-visible` instead */
  tagInputRootFocused: {},
  /** @deprecated unused */
  tagInputRootEmpty: {},
  singleLine: {},
  error: { float: "left" },
  inputExtension: {},
  suggestionsContainer: {},
  suggestionList: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
