// @hitachivantara/uikit-react-core/dist/cjs/Dialog/Actions/Actions.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const MuiDialogActions = require("@mui/material/DialogActions");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const context = require("../context.cjs");
const Actions_styles = require("./Actions.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const MuiDialogActions__default = /* @__PURE__ */ _interopDefault(MuiDialogActions);
const HvDialogActions = (props) => {
  const {
    classes: classesProp,
    className,
    children,
    fullscreen: fullScreenProp,
    ...others
  } = uikitReactUtils.useDefaultProps("HvDialogActions", props);
  const context$1 = context.useDialogContext();
  const fullScreen = fullScreenProp ?? context$1.fullScreen;
  const { classes, cx } = Actions_styles.useClasses(classesProp);
  return /* @__PURE__ */ jsxRuntime.jsx(
    MuiDialogActions__default.default,
    {
      className,
      classes: {
        root: cx(classes.root, { [classes.fullscreen]: fullScreen }),
        spacing: classes.spacing
      },
      ...others,
      children
    }
  );
};
exports.dialogActionClasses = Actions_styles.staticClasses;
exports.HvDialogActions = HvDialogActions;
