// @hitachivantara/uikit-react-core/dist/cjs/CheckBoxGroup/CheckBoxGroup.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvCheckBoxGroup", {
  root: {
    display: "inline-block",
    overflow: "clip",
    overflowClipMargin: 4,
    verticalAlign: "top"
  },
  label: {},
  group: { display: "flex" },
  vertical: {
    flexDirection: "column",
    // Prevent the focus ring to be hidden by sibling hover background
    "&>*": {
      zIndex: 0
    },
    "&>*:focus-within": {
      zIndex: 1
    }
  },
  horizontal: {
    flexDirection: "row",
    flexWrap: "wrap",
    gap: uikitStyles.theme.space.sm
  },
  invalid: {
    paddingBottom: uikitStyles.theme.space.xs,
    borderBottom: `1px solid ${uikitStyles.theme.colors.negative_120}`
  },
  selectAll: {},
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
