// @hitachivantara/uikit-react-core/dist/cjs/MultiButton/MultiButton.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const DropDownMenu_styles = require("../DropDownMenu/DropDownMenu.styles.cjs");
require("../DropDownMenu/DropDownMenu.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvMultiButton", {
  root: {
    display: "flex",
    alignItems: "center",
    position: "relative"
  },
  multiple: {
    backgroundColor: uikitStyles.theme.colors.atmo2,
    borderWidth: 0,
    borderColor: `${uikitStyles.theme.colors.atmo4} transparent`,
    borderRadius: uikitStyles.theme.radii.base,
    "& $button": {
      minWidth: 32,
      width: "100%",
      maxWidth: 200,
      padding: 0,
      flex: "1 1 auto",
      borderColor: "inherit",
      borderRadius: 0,
      fontWeight: uikitStyles.theme.typography.body.fontWeight,
      "&:disabled": {
        color: uikitStyles.theme.colors.secondary_60,
        borderColor: "inherit"
      },
      "&:hover": {
        borderColor: "inherit"
      },
      "&$firstButton": {
        borderLeftColor: uikitStyles.theme.colors.atmo4,
        borderTopLeftRadius: "inherit",
        borderBottomLeftRadius: "inherit",
        "&:disabled": {
          borderLeftColor: uikitStyles.theme.colors.atmo4
        }
      },
      "&$lastButton": {
        borderRightColor: uikitStyles.theme.colors.atmo4,
        borderTopRightRadius: "inherit",
        borderBottomRightRadius: "inherit",
        "&:disabled": {
          borderRightColor: uikitStyles.theme.colors.atmo4
        }
      },
      "&:not($firstButton)": {
        marginLeft: "-1px"
      },
      "&$selected": {
        backgroundColor: uikitStyles.theme.colors.atmo1,
        fontWeight: uikitStyles.theme.typography.label.fontWeight,
        borderRadius: "inherit",
        borderColor: uikitStyles.theme.colors.secondary,
        zIndex: 2,
        "&:hover:not(:disabled),&:focus-visible": {
          backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
        },
        "&:disabled": {
          zIndex: 1,
          borderColor: uikitStyles.theme.colors.atmo4
        }
      }
    }
  },
  splitGroup: {
    width: "fit-content",
    "& $button:not($firstButton)": {
      borderTopLeftRadius: 0,
      borderBottomLeftRadius: 0,
      "&:not([aria-controls])": {
        borderLeftWidth: 0
      }
    },
    "& $button:not($lastButton)": {
      borderTopRightRadius: 0,
      borderBottomRightRadius: 0,
      "&:not([aria-controls])": {
        borderRightWidth: 0
      },
      "&::after": {
        content: "''",
        position: "absolute",
        inset: "4px -1px 4px auto",
        width: 1,
        zIndex: 1,
        height: "auto",
        backgroundColor: "currentcolor"
      }
    }
  },
  splitGroupDisabled: {},
  button: {
    position: "relative",
    // prevent the focus ring to be hidden by sibling hover background
    [`&:focus-visible, &.${DropDownMenu_styles.staticClasses.iconSelected}`]: {
      zIndex: 5
    }
  },
  selected: {},
  vertical: {
    flexDirection: "column",
    height: "auto",
    borderColor: `transparent ${uikitStyles.theme.colors.atmo4}`,
    "& $button": {
      minWidth: 32,
      width: "100%",
      "&$firstButton": {
        borderTopColor: uikitStyles.theme.colors.atmo4,
        borderTopLeftRadius: "inherit",
        borderTopRightRadius: "inherit"
      },
      "&$lastButton": {
        borderBottomColor: uikitStyles.theme.colors.atmo4,
        borderBottomLeftRadius: "inherit",
        borderBottomRightRadius: "inherit"
      },
      "&:not($firstButton)": {
        marginLeft: 0,
        marginTop: -1
      },
      "&$selected": {
        height: 32,
        width: "calc(100% + 2px)",
        borderColor: uikitStyles.theme.colors.secondary
      }
    }
  },
  // TODO - review the need for these classes in v6 (use :first-child and :last-child instead)
  firstButton: {},
  lastButton: {},
  // TODO - review the need for these classes in v6
  primary: {},
  primarySubtle: {},
  primaryGhost: {},
  secondary: {},
  secondarySubtle: {},
  secondaryGhost: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
