// @hitachivantara/uikit-react-core/dist/cjs/Select/OptionGroup.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const OptionGroup = require("@mui/base/OptionGroup");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvOptionGroup", {
  root: {
    listStyle: "none",
    ...uikitStyles.theme.typography.label
  }
});
const HvOptionGroup = React.forwardRef(
  function HvOptionGroup2(props, ref) {
    const {
      className,
      classes: classesProp,
      ...others
    } = uikitReactUtils.useDefaultProps("HvOptionGroup", props);
    const { classes, cx } = useClasses(classesProp);
    return /* @__PURE__ */ jsxRuntime.jsx(
      OptionGroup.OptionGroup,
      {
        ref,
        className: cx(classes.root, className),
        ...others
      }
    );
  }
);
exports.HvOptionGroup = HvOptionGroup;
exports.optionGroupClasses = staticClasses;
