// @hitachivantara/uikit-react-core/dist/cjs/Select/Select.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvSelect", {
  root: {
    position: "relative",
    "&$disabled,&$readOnly": {
      pointerEvents: "none"
    }
  },
  disabled: {},
  readOnly: {},
  invalid: {
    borderColor: uikitStyles.theme.colors.negative_120
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-start"
  },
  label: {},
  description: {},
  select: {},
  popper: {
    zIndex: uikitStyles.theme.zIndices.popover
  },
  panel: {
    maxHeight: 400,
    border: `1px solid ${uikitStyles.theme.colors.secondary}`,
    marginTop: -1,
    marginBottom: -1,
    // panel styles
    position: "relative",
    padding: uikitStyles.theme.space.xs,
    backgroundColor: uikitStyles.theme.colors.atmo1,
    overflowY: "auto",
    borderRadius: "inherit"
  },
  panelOpenedUp: {
    borderRadius: `${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base} 0 0`
  },
  panelOpenedDown: {
    borderRadius: `0 0 ${uikitStyles.theme.radii.base} ${uikitStyles.theme.radii.base}`
  },
  error: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
