// @hitachivantara/uikit-react-core/dist/cjs/Calendar/SingleCalendar/CalendarCell.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const hover = {
  backgroundColor: uikitStyles.theme.colors.containerBackgroundHover,
  cursor: "pointer"
};
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvCalendarCell", {
  cellContainer: {},
  calendarDate: {
    display: "flex",
    justifyContent: "center",
    flexDirection: "column",
    textAlign: "center",
    height: "40px",
    width: "40px"
  },
  calendarDateSelected: {
    backgroundColor: uikitStyles.theme.colors.atmo3,
    color: uikitStyles.theme.colors.secondary,
    "&:hover": {
      ...hover
    }
  },
  calendarDateNotInMonth: {
    color: uikitStyles.theme.colors.secondary_60,
    cursor: "not-allowed"
  },
  calendarDateInSelectionRange: {
    backgroundColor: uikitStyles.theme.colors.atmo3
  },
  calendarDateDisabled: {
    color: uikitStyles.theme.colors.secondary_60,
    cursor: "no-drop"
  },
  startBookend: {
    borderLeft: `1px solid ${uikitStyles.theme.colors.secondary}`
  },
  endBookend: {
    borderRight: `1px solid ${uikitStyles.theme.colors.secondary}`
  },
  dateWrapper: {
    width: "40px",
    height: "40px"
  },
  cellsInRange: {
    "&:focus": {
      outside: "none"
    }
  },
  cellsOutsideRange: {
    "&:focus": {
      outside: "none"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
