// @hitachivantara/uikit-react-core/dist/cjs/ButtonBase/ButtonBase.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvButtonBase", {
  root: {
    display: "inline-flex",
    cursor: "pointer",
    background: "none",
    padding: 0,
    // Background color common for almost all variants
    ":where(:not($disabled))": {
      ":hover, :focus-visible": {
        backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
      }
    },
    ":focus-visible": {
      ...focusUtils.outlineStyles
    },
    // Default button - no size specified
    fontFamily: uikitStyles.theme.fontFamily.body,
    fontSize: "inherit",
    color: "inherit"
  },
  disabled: {
    cursor: "not-allowed",
    color: uikitStyles.theme.colors.secondary_60,
    "&:hover, &:focus-visible": {
      backgroundColor: "transparent"
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
