// @hitachivantara/uikit-react-core/dist/cjs/FormElement/InfoMessage/InfoMessage.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const setId = require("../../utils/setId.cjs");
const context = require("../context.cjs");
const InfoMessage_styles = require("./InfoMessage.styles.cjs");
const Typography = require("../../Typography/Typography.cjs");
const HvInfoMessage = (props) => {
  const {
    id: idProp,
    classes: classesProp,
    className,
    children,
    disabled: disabledProp,
    disableGutter = false,
    ...others
  } = uikitReactUtils.useDefaultProps("HvInfoMessage", props);
  const { classes, cx } = InfoMessage_styles.useClasses(classesProp);
  const context$1 = React.useContext(context.HvFormElementContext);
  const disabled = disabledProp ?? context$1.disabled;
  const id = idProp ?? setId.setId(context$1.id, "description");
  return /* @__PURE__ */ jsxRuntime.jsx(
    Typography.HvTypography,
    {
      id,
      className: cx(
        classes.root,
        {
          [classes.infoDisabled]: !!disabled,
          [classes.gutter]: !disableGutter
        },
        className
      ),
      variant: "body",
      component: "label",
      ...others,
      children
    }
  );
};
exports.infoMessageClasses = InfoMessage_styles.staticClasses;
exports.HvInfoMessage = HvInfoMessage;
