// @hitachivantara/uikit-react-core/dist/cjs/QueryBuilder/QueryBuilder.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { useClasses, staticClasses } = uikitReactUtils.createClasses("HvQueryBuilder", {
  /** Styles applied to the component root class. */
  root: {
    position: "relative",
    padding: uikitStyles.theme.space.sm,
    marginTop: 12,
    marginBottom: 32,
    border: `1px solid ${uikitStyles.theme.colors.atmo4}`
  },
  /** Styles applied to the top group container. */
  topGroup: {
    margin: uikitStyles.theme.space.sm,
    backgroundColor: "transparent",
    maxWidth: "100%",
    paddingBottom: uikitStyles.theme.space.lg
  },
  /** Styles applied to the sub group containers. */
  subGroup: {
    margin: "40px 14px 32px 20px",
    minHeight: 120,
    paddingBottom: uikitStyles.theme.space.md,
    "&::before": {
      content: '""',
      position: "absolute",
      zIndex: 2,
      width: 21,
      height: 36,
      borderBottom: `1px solid ${uikitStyles.theme.colors.atmo4}`,
      borderLeft: `1px solid ${uikitStyles.theme.colors.atmo4}`,
      top: -38,
      left: -38
    },
    ":not($topRulesContainer)>&:last-child::after": {
      content: '""',
      position: "absolute",
      zIndex: 1,
      width: 32,
      height: "100%",
      borderLeft: `1px solid ${uikitStyles.theme.colors.atmo4}`,
      top: 0,
      left: `calc(${uikitStyles.theme.space.sm} + 2)`
    }
  },
  /** Styles applied to the radio button+label topGroup when the radio button is disabled. */
  combinator: { minWidth: 80 },
  /** Styles applied to the multi-button combinator container on the top group. */
  topCombinator: {
    position: "absolute",
    top: `calc(-1 * ${uikitStyles.theme.space.sm})`,
    left: `calc(-1 * ${uikitStyles.theme.space.sm})`
  },
  /** Styles applied to each combinator button. */
  combinatorButton: {},
  /** Styles applied to the remove button.  */
  removeButton: {},
  /** Styles applied to the remove button on the top group. */
  topRemoveButton: { position: "absolute", top: -16, right: -16 },
  /** Styles applied to the remove button when disabled on the top group. */
  topRemoveButtonDisabled: { backgroundColor: uikitStyles.theme.colors.atmo2 },
  /** Styles applied to the rules container. */
  rulesContainer: {},
  /** Styles applied to the sub rules container. */
  subRulesContainer: {
    borderLeft: `1px solid ${uikitStyles.theme.colors.atmo4}`,
    marginLeft: uikitStyles.theme.space.sm,
    marginBottom: uikitStyles.theme.space.md,
    paddingLeft: uikitStyles.theme.space.sm,
    paddingTop: 7,
    position: "relative",
    left: -33,
    width: "100%"
  },
  /** Styles applied to the action button container. */
  actionButtonContainer: {
    marginLeft: "auto",
    "&>*": {
      marginLeft: uikitStyles.theme.space.sm
    }
  },
  /** Styles applied to the top action button container. */
  topActionButtonContainer: {
    position: "absolute",
    bottom: `calc(-1 * ${uikitStyles.theme.space.md} * 0.5 - 3px)`,
    right: `calc(${uikitStyles.theme.space.sm} * 1.75 + 2px)`
  },
  /** Styles applied to the top rules container. */
  topRulesContainer: {},
  /** Styles applied to the background of buttons to remove transparency */
  buttonBackground: {
    backgroundColor: uikitStyles.theme.colors.atmo2,
    display: "inline-flex"
  },
  createConditionButton: {
    cursor: "pointer",
    backgroundColor: "transparent",
    padding: 0,
    "&:disabled": { cursor: "not-allowed", pointerEvents: "none" }
  },
  createGroupButton: {
    cursor: "pointer",
    backgroundColor: "transparent",
    padding: 0,
    "&:disabled": { cursor: "not-allowed", pointerEvents: "none" }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
