// @hitachivantara/uikit-react-core/dist/cjs/Tooltip/Tooltip.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const Tooltip = require("@mui/material/Tooltip");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvTooltip", {
  root: {},
  tooltip: {},
  popper: {
    // TODO: move this to `tooltip` in v6 and make class always be applied
    [`& .${Tooltip.tooltipClasses.tooltip}`]: {
      ...uikitStyles.theme.typography.body,
      display: "flex",
      width: "fit-content",
      maxWidth: 532,
      backgroundColor: uikitStyles.theme.colors.atmo1,
      boxShadow: uikitStyles.theme.colors.shadow,
      padding: 0,
      "& p": {
        display: "-webkit-box",
        width: "fit-content",
        boxOrient: "vertical",
        textOverflow: "ellipsis",
        overflow: "hidden"
      }
    },
    "& $title": {
      padding: "15px 20px",
      borderBottom: `3px solid ${uikitStyles.theme.colors.atmo2}`
    },
    "& $valuesContainer": {
      padding: uikitStyles.theme.space.sm
    },
    "& $values": {
      display: "flex",
      justifyContent: "space-between",
      paddingBottom: "10px",
      "&:last-child": {
        paddingBottom: 0
      }
    },
    "& $color": {
      width: uikitStyles.theme.space.xs
    },
    "& $separator": {
      width: uikitStyles.theme.space.xs
    },
    "& $separatorColor": {
      width: "5px"
    },
    "& $valueWrapper": {
      padding: uikitStyles.theme.space.sm
    },
    "& $tooltip": {
      padding: uikitStyles.theme.space.sm,
      borderRadius: uikitStyles.theme.radii.round,
      "& p": {
        wordBreak: "break-word"
      }
    }
  },
  // TODO: remove in v6
  /** @deprecated */
  tooltipMulti: {},
  /** @deprecated */
  title: {},
  /** @deprecated */
  valuesContainer: {},
  /** @deprecated */
  values: {},
  /** @deprecated */
  color: {},
  /** @deprecated */
  separatorColor: {},
  /** @deprecated */
  separator: {},
  /** @deprecated */
  valueWrapper: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
