// @hitachivantara/uikit-react-core/dist/cjs/Tooltip/Tooltip.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const Fade = require("@mui/material/Fade");
const Tooltip = require("@mui/material/Tooltip");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const document = require("../utils/document.cjs");
const Tooltip_styles = require("./Tooltip.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const Fade__default = /* @__PURE__ */ _interopDefault(Fade);
const Tooltip__default = /* @__PURE__ */ _interopDefault(Tooltip);
const HvTooltip = React.forwardRef(function HvTooltip2(props, ref) {
  const {
    className,
    classes: classesProp,
    open,
    enterDelay = 300,
    placement = "top",
    useSingle = true,
    children,
    title,
    TransitionComponent = Fade__default.default,
    TransitionProps = { timeout: 400, placement: "top" },
    containerId,
    ...others
  } = uikitReactUtils.useDefaultProps("HvTooltip", props);
  const { rootId } = uikitReactUtils.useTheme();
  const { classes } = Tooltip_styles.useClasses(classesProp);
  const [container, setContainer] = React.useState(
    () => document.getElementById(containerId ?? rootId)
  );
  React.useEffect(() => {
    setContainer(document.getElementById(containerId ?? rootId));
  }, [containerId, rootId]);
  return /* @__PURE__ */ jsxRuntime.jsx(
    Tooltip__default.default,
    {
      ref,
      open,
      enterDelay,
      placement,
      TransitionComponent,
      TransitionProps,
      className,
      classes: {
        tooltip: useSingle ? classes.tooltip : classes.tooltipMulti,
        popper: classes.popper
      },
      title,
      PopperProps: { container },
      ...others,
      children
    }
  );
});
exports.tooltipClasses = Tooltip_styles.staticClasses;
exports.HvTooltip = HvTooltip;
