// @hitachivantara/uikit-react-core/dist/cjs/VerticalNavigation/NavigationPopup/NavigationPopupContainer.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const Popper = require("@mui/base/Popper");
const ClickAwayListener = require("@mui/material/ClickAwayListener");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const document = require("../../utils/document.cjs");
const VerticalNavigation = require("../VerticalNavigation.cjs");
const NavigationPopup_styles = require("./NavigationPopup.styles.cjs");
const _interopDefault = (e) => e && e.__esModule ? e : { default: e };
const ClickAwayListener__default = /* @__PURE__ */ _interopDefault(ClickAwayListener);
const NavigationPopupContainer = ({
  anchorEl,
  onClose,
  children,
  classes: classesProp,
  className,
  ...others
}) => {
  const { classes, cx } = NavigationPopup_styles.useClasses(classesProp);
  const { rootId } = uikitReactUtils.useTheme();
  const handleClickAway = () => onClose?.();
  return /* @__PURE__ */ jsxRuntime.jsx(
    Popper.Popper,
    {
      open: true,
      anchorEl,
      placement: "right-start",
      container: document.getContainerElement(rootId),
      className: cx(classes.popper, classes.popup, className),
      ...others,
      children: /* @__PURE__ */ jsxRuntime.jsx(ClickAwayListener__default.default, { onClickAway: handleClickAway, children: /* @__PURE__ */ jsxRuntime.jsx("div", { className: classes.container, children: /* @__PURE__ */ jsxRuntime.jsx(VerticalNavigation.HvVerticalNavigation, { open: true, useIcons: true, children }) }) })
    }
  );
};
exports.verticalNavigationPopupClasses = NavigationPopup_styles.staticClasses;
exports.NavigationPopupContainer = NavigationPopupContainer;
