// @hitachivantara/uikit-react-core/dist/esm/DropdownButton/DropdownButton.js
import { jsx } from "react/jsx-runtime";
import { forwardRef } from "react";
import { DropDownXS } from "@hitachivantara/uikit-react-icons";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./DropdownButton.styles.js";
import { staticClasses } from "./DropdownButton.styles.js";
import { HvButton } from "../Button/Button.js";
const HvDropdownButton = forwardRef(function HvDropdownButton2(props, ref) {
  const {
    className,
    classes: classesProp,
    placement = "bottom",
    disabled,
    open,
    icon,
    readOnly,
    children,
    variant,
    ...others
  } = useDefaultProps("HvDropdownButton", props);
  const { classes, cx } = useClasses(classesProp);
  const endIcon = icon ? void 0 : /* @__PURE__ */ jsx(DropDownXS, { size: "XS", rotate: open });
  return /* @__PURE__ */ jsx(
    HvButton,
    {
      ref,
      icon,
      disabled: disabled || readOnly,
      className: cx(
        classes.root,
        {
          [classes.open]: open,
          [classes.openUp]: open && placement.includes("top"),
          [classes.openDown]: open && placement.includes("bottom"),
          [classes.disabled]: disabled,
          [classes.readOnly]: readOnly
        },
        className
      ),
      classes: { endIcon: classes.arrowContainer },
      endIcon,
      variant: open ? "secondarySubtle" : variant,
      ...others,
      children: /* @__PURE__ */ jsx("div", { className: cx({ [classes.selection]: !icon }), children: children && typeof children === "string" ? /* @__PURE__ */ jsx("div", { className: classes.placeholder, children }) : children })
    }
  );
});
export {
  HvDropdownButton,
  staticClasses as dropdownButtonClasses
};
