// @hitachivantara/uikit-react-core/dist/esm/BaseRadio/BaseRadio.js
import { jsx } from "react/jsx-runtime";
import { forwardRef, useState, useCallback } from "react";
import MuiRadio from "@mui/material/Radio";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./BaseRadio.styles.js";
import { staticClasses } from "./BaseRadio.styles.js";
import { Selected, Unselected } from "./icons.js";
const getSelectorIcons = () => {
  return {
    radio: /* @__PURE__ */ jsx(Unselected, {}),
    radioChecked: /* @__PURE__ */ jsx(Selected, {})
  };
};
const HvBaseRadio = forwardRef(
  function HvBaseRadio2(props, ref) {
    const {
      classes: classesProp,
      className,
      id,
      name,
      value = "on",
      required = false,
      readOnly = false,
      disabled = false,
      checked,
      defaultChecked,
      onChange,
      semantic = false,
      inputProps,
      onFocusVisible,
      onBlur,
      icon,
      checkedIcon,
      ...others
    } = useDefaultProps("HvBaseRadio", props);
    const { classes, cx } = useClasses(classesProp);
    const [focusVisible, setFocusVisible] = useState(false);
    const onFocusVisibleCallback = useCallback(
      (evt) => {
        setFocusVisible(true);
        onFocusVisible?.(evt);
      },
      [onFocusVisible]
    );
    const onBlurCallback = useCallback(
      (evt) => {
        setFocusVisible(false);
        onBlur?.(evt);
      },
      [onBlur]
    );
    const icons = getSelectorIcons();
    const onLocalChange = useCallback(
      (evt) => {
        if (readOnly) {
          return;
        }
        onChange?.(evt, evt.target.checked, value);
      },
      [onChange, readOnly, value]
    );
    return /* @__PURE__ */ jsx(
      MuiRadio,
      {
        ref,
        id,
        name,
        className: cx(
          classes.root,
          {
            [classes.disabled]: disabled,
            [classes.focusVisible]: focusVisible,
            [classes.checked]: checked,
            [classes.semantic]: semantic
          },
          className
        ),
        icon: icon || icons.radio,
        checkedIcon: checkedIcon || icons.radioChecked,
        color: "default",
        disabled,
        required,
        readOnly,
        onChange: onLocalChange,
        value,
        checked,
        defaultChecked,
        inputProps,
        onFocusVisible: onFocusVisibleCallback,
        onBlur: onBlurCallback,
        ...others
      }
    );
  }
);
export {
  HvBaseRadio,
  staticClasses as baseRadioClasses,
  getSelectorIcons
};
