// @hitachivantara/uikit-react-core/dist/cjs/Radio/Radio.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvRadio", {
  root: { display: "inline-block" },
  container: {
    cursor: "pointer",
    display: "flex",
    alignItems: "center",
    transition: "background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
    borderRadius: uikitStyles.theme.radii.base,
    ":hover:not($disabled)": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
    }
  },
  invalidContainer: {},
  disabled: {
    cursor: "not-allowed",
    "& $label": { color: uikitStyles.theme.colors.secondary_60, cursor: "not-allowed" }
  },
  radio: {
    "& svg": {
      outline: "none",
      boxShadow: "none"
    }
  },
  invalidRadio: {},
  label: {
    verticalAlign: "middle",
    paddingRight: uikitStyles.theme.space.xs,
    ...uikitStyles.theme.typography.body,
    cursor: "pointer",
    lineHeight: "32px",
    width: "100%"
  },
  focusVisible: { backgroundColor: uikitStyles.theme.colors.atmo3, ...focusUtils.outlineStyles },
  checked: {},
  semantic: {}
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
