// @hitachivantara/uikit-react-core/dist/cjs/QueryBuilder/Rule/Rule.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const QueryBuilder_styles = require("../QueryBuilder.styles.cjs");
const { useClasses, staticClasses } = uikitReactUtils.createClasses(
  "HvQueryBuilder-Rule",
  {
    root: {
      position: "relative",
      marginTop: uikitStyles.theme.space.xs,
      minHeight: 94,
      "&>div:not(:last-child)": {
        paddingRight: uikitStyles.theme.space.md
      },
      // hide required * as all fields are required
      "& label>span[aria-hidden]": {
        visibility: "hidden"
      },
      "&::before": {
        content: '""',
        position: "absolute",
        zIndex: 2,
        width: "17px",
        height: "39px",
        borderBottom: `1px solid ${uikitStyles.theme.colors.atmo4}`,
        borderLeft: `1px solid ${uikitStyles.theme.colors.atmo4}`,
        top: 0,
        left: `calc(-1 * 17px)`
      },
      [`:not(.${QueryBuilder_styles.staticClasses.topRulesContainer})>&:last-child::after`]: {
        content: '""',
        position: "absolute",
        zIndex: 1,
        width: "17px",
        height: "100%",
        borderLeft: `1px solid ${uikitStyles.theme.colors.atmo4}`,
        top: 0,
        left: `calc(-1 * 17px)`
      }
    },
    actionsContainer: {
      marginLeft: "auto",
      marginTop: "24px",
      "&>:not(:last-child)": {
        marginRight: uikitStyles.theme.space.xs
      }
    },
    isMdDown: {
      "&>div:not(:last-child)": {
        paddingRight: 0
      },
      "&>div:not(:first-of-type)": {
        marginTop: uikitStyles.theme.space.xs
      }
    }
  }
);
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
