// @hitachivantara/uikit-react-core/dist/cjs/TreeView/internals/TreeViewProvider.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const DescendantProvider = require("./DescendantProvider.cjs");
const DEFAULT_TREE_VIEW_CONTEXT_VALUE = {
  instance: null,
  multiSelect: false,
  disabledItemsFocusable: false,
  treeId: void 0,
  icons: {
    defaultCollapseIcon: null,
    defaultExpandIcon: null,
    defaultParentIcon: null,
    defaultEndIcon: null
  }
};
const TreeViewContext = React.createContext(
  DEFAULT_TREE_VIEW_CONTEXT_VALUE
);
const TreeViewProvider = (props) => {
  const { value, children } = props;
  return /* @__PURE__ */ jsxRuntime.jsx(TreeViewContext.Provider, { value, children: /* @__PURE__ */ jsxRuntime.jsx(DescendantProvider.DescendantProvider, { children }) });
};
const useTreeViewContext = () => React.useContext(TreeViewContext);
exports.DEFAULT_TREE_VIEW_CONTEXT_VALUE = DEFAULT_TREE_VIEW_CONTEXT_VALUE;
exports.TreeViewContext = TreeViewContext;
exports.TreeViewProvider = TreeViewProvider;
exports.useTreeViewContext = useTreeViewContext;
