// @hitachivantara/uikit-react-core/dist/esm/Card/Card.js
import { jsxs, jsx } from "react/jsx-runtime";
import { forwardRef } from "react";
import { useDefaultProps, mergeStyles } from "@hitachivantara/uikit-react-utils";
import { getColor } from "@hitachivantara/uikit-styles";
import { useClasses } from "./Card.styles.js";
import { staticClasses } from "./Card.styles.js";
const HvCard = forwardRef(function HvCard2(props, ref) {
  const {
    classes: classesProp,
    style,
    className,
    children,
    icon,
    selectable = false,
    selected = false,
    statusColor = "sema0",
    bgcolor,
    ...others
  } = useDefaultProps("HvCard", props);
  const { classes, cx } = useClasses(classesProp);
  const barColor = statusColor !== "sema0" && statusColor || selected && "secondary" || "atmo4";
  return /* @__PURE__ */ jsxs(
    "div",
    {
      ref,
      style: mergeStyles(style, {
        "--bg-color": getColor(bgcolor),
        "--bar-height": `${selected ? 4 : 2}px`,
        "--bar-color": getColor(barColor)
      }),
      className: cx(
        "HvIsCardGridElement",
        classes.root,
        {
          [classes.selectable]: selectable,
          [classes.selected]: selected
        },
        className
      ),
      ...others,
      children: [
        /* @__PURE__ */ jsxs("div", { className: classes.semanticContainer, children: [
          /* @__PURE__ */ jsx("div", { className: classes.semanticBar }),
          /* @__PURE__ */ jsx("div", { className: classes.icon, children: icon })
        ] }),
        children
      ]
    }
  );
});
export {
  HvCard,
  staticClasses as cardClasses
};
