// @hitachivantara/uikit-react-core/dist/cjs/Button/Button.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvButton", {
  /** applied to the root element */
  root: {
    display: "inline-flex",
    alignItems: "center",
    justifyContent: "center",
    textTransform: "none",
    cursor: "pointer",
    whiteSpace: "nowrap",
    // Background color common for almost all variants
    ":where(:not($disabled))": {
      ":hover, :focus-visible": {
        backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
      }
    },
    ":focus-visible": {
      ...focusUtils.outlineStyles
    },
    // Default button - no size specified
    fontFamily: uikitStyles.theme.fontFamily.body,
    ...uikitStyles.theme.typography.label,
    color: "var(--color, currentcolor)",
    backgroundColor: "transparent",
    height: "var(--HvButton-height)",
    border: "1px solid transparent",
    borderRadius: `var(--radius, ${uikitStyles.theme.radii.base})`,
    padding: uikitStyles.theme.spacing(0, "sm")
  },
  /** applied to the _left_ icon container */
  startIcon: {
    marginLeft: uikitStyles.theme.spacing(-1),
    marginTop: -1,
    marginBottom: -1
  },
  /** applied to the _right_ icon container */
  endIcon: {
    marginRight: uikitStyles.theme.spacing(-1),
    marginTop: -1,
    marginBottom: -1
  },
  focusVisible: {},
  /** applied to the root element when disabled */
  disabled: {
    cursor: "not-allowed",
    color: uikitStyles.theme.colors.secondary_60,
    backgroundColor: "transparent",
    borderColor: "transparent",
    ":hover, :focus-visible": {
      backgroundColor: "transparent",
      borderColor: "transparent"
    }
  },
  /** applied to the root element when is icon-only */
  icon: {
    margin: 0,
    padding: 0,
    height: "fit-content",
    "& > *": {
      margin: -1
    }
  },
  /** applied to the root element when using the `contained` variant */
  contained: {
    color: uikitStyles.theme.colors.atmo1,
    // `color-contrast(var(--color) vs ${colors.atmo1}, ${colors.base_light}, ${colors.base_dark})`,
    backgroundColor: "var(--color)",
    ":where(:not($disabled))": {
      ":hover, :focus-visible": {
        backgroundColor: "color-mix(in srgb, var(--color), black 20%)"
      },
      ":active": {
        backgroundColor: "color-mix(in srgb, var(--color), black 30%)"
      }
    }
  },
  /** applied to the root element when using the `subtle` variant */
  subtle: {
    borderColor: "currentcolor"
  },
  /** applied to the root element when using the `ghost` variant */
  ghost: {},
  /** applied to the root element when using the `semantic` variant */
  semantic: {
    color: uikitStyles.theme.colors.base_dark,
    backgroundColor: "transparent",
    borderColor: "transparent",
    "&:hover, &:focus-visible": {
      backgroundColor: uikitStyles.theme.alpha("base_light", 0.3)
    },
    "&:active": {
      backgroundColor: uikitStyles.theme.alpha("base_light", 0.5)
    },
    "&$disabled": {
      backgroundColor: uikitStyles.theme.alpha("base_light", 0.1)
    }
  },
  // TODO - remove in v6
  primary: {},
  primarySubtle: {},
  primaryGhost: {},
  secondarySubtle: {},
  secondaryGhost: {},
  // Deprecated (DS3)
  secondary: {}
});
const sizes = {
  xs: { height: "24px", fontSize: "sm" },
  sm: { height: "24px", fontSize: "sm" },
  md: { height: "32px" },
  lg: { height: "48px", space: "md" },
  xl: { height: "48px", space: "md" }
};
const getSizeStyles = (size) => {
  const { height, space = "sm", fontSize } = sizes[size];
  return {
    height,
    padding: uikitStyles.theme.spacing(0, space),
    fontSize: fontSize && uikitStyles.theme.fontSizes[fontSize]
  };
};
const getIconSizeStyles = (size) => {
  const { height } = sizes[size];
  return {
    height,
    width: height
  };
};
const getOverrideColors = () => ({
  "& svg .color0": {
    fill: "currentcolor"
  }
});
exports.getIconSizeStyles = getIconSizeStyles;
exports.getOverrideColors = getOverrideColors;
exports.getSizeStyles = getSizeStyles;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
