// @hitachivantara/uikit-react-core/dist/cjs/SelectionList/SelectionList.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvSelectionList", {
  root: { display: "inline-block", padding: 0, margin: 0 },
  error: { width: "100%", float: "left", clear: "both" },
  listbox: {
    display: "flex",
    float: "left",
    clear: "both",
    width: "100%",
    paddingBottom: uikitStyles.theme.space.xs
  },
  label: {},
  description: { float: "left" },
  horizontal: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginLeft: -uikitStyles.theme.space.sm,
    "&>*": {
      marginLeft: uikitStyles.theme.space.sm
    }
  },
  vertical: {
    flexDirection: "column",
    // Prevent the focus ring to be hidden by sibling hover background
    "&>*": {
      zIndex: 0
    },
    "&>*:focus-within": {
      zIndex: 1
    }
  },
  invalid: { borderBottom: `1px solid ${uikitStyles.theme.colors.negative_120}` }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
