// @hitachivantara/uikit-react-core/dist/esm/VerticalNavigation/Actions/Action.js
import { jsxs } from "react/jsx-runtime";
import { useContext } from "react";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { setId } from "../../utils/setId.js";
import { VerticalNavigationContext } from "../VerticalNavigationContext.js";
import { useClasses } from "./Action.styles.js";
import { staticClasses } from "./Action.styles.js";
import { HvButton } from "../../Button/Button.js";
const HvVerticalNavigationAction = (props) => {
  const {
    className,
    classes: classesProp,
    id,
    label = "",
    icon,
    ...others
  } = useDefaultProps("HvVerticalNavigationAction", props);
  const { isOpen } = useContext(VerticalNavigationContext);
  const { classes, cx } = useClasses(classesProp);
  return /* @__PURE__ */ jsxs(
    HvButton,
    {
      id: setId(id, "button"),
      variant: "secondaryGhost",
      icon: !isOpen,
      className: cx(
        classes.action,
        {
          [classes.noIcon]: !icon,
          [classes.minimized]: !isOpen
        },
        className
      ),
      ...!isOpen && { "aria-label": label },
      ...others,
      children: [
        icon,
        isOpen && label
      ]
    }
  );
};
export {
  HvVerticalNavigationAction,
  staticClasses as actionClasses
};
