// @hitachivantara/uikit-react-core/dist/cjs/ScrollToHorizontal/HorizontalScrollListItem/HorizontalScrollListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const name = "HvHorizontalScrollListItem";
const { staticClasses, useClasses } = uikitReactUtils.createClasses(name, {
  root: {
    padding: uikitStyles.theme.spacing("xs", 0),
    maxWidth: 120
  },
  button: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    }
  },
  text: {
    margin: uikitStyles.theme.spacing("xs", "xs", "0")
  },
  selected: {
    fontWeight: uikitStyles.theme.typography.label.fontWeight
  },
  bullet: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    minHeight: 24,
    height: 24,
    width: 24,
    fontSize: 4,
    color: uikitStyles.theme.colors.secondary_60,
    "& > span": {
      margin: "auto",
      width: "1em",
      height: "1em",
      backgroundColor: "currentcolor",
      borderRadius: "50%"
    }
  },
  bulletSelected: {
    fontSize: 6,
    color: uikitStyles.theme.colors.secondary
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
