// @hitachivantara/uikit-react-core/dist/cjs/FilterGroup/FilterGroup.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const useControlled = require("../hooks/useControlled.cjs");
const useLabels = require("../hooks/useLabels.cjs");
const useUniqueId = require("../hooks/useUniqueId.cjs");
const setId = require("../utils/setId.cjs");
const FilterGroup_styles = require("./FilterGroup.styles.cjs");
const FilterGroupContext = require("./FilterGroupContext.cjs");
const FilterContent = require("./FilterContent/FilterContent.cjs");
const FormElement = require("../FormElement/FormElement.cjs");
const Label = require("../FormElement/Label/Label.cjs");
const InfoMessage = require("../FormElement/InfoMessage/InfoMessage.cjs");
const WarningText = require("../FormElement/WarningText/WarningText.cjs");
const DEFAULT_LABELS = {
  /** Apply button label. */
  applyLabel: "Apply",
  /** Cancel button label. */
  cancelLabel: "Cancel",
  /** Clear button label. */
  clearLabel: "Clear Filters",
  /** Placeholder label. */
  placeholder: "Filters",
  /** SearchBox placeholder label. */
  searchBoxPlaceholder: "Search",
  /** Select All placeholder label. */
  selectAll: "All",
  /** Multi selection conjunction placeholder label. */
  multiSelectionConjunction: "/"
};
const HvFilterGroup = React.forwardRef(
  function HvFilterGroup2(props, ref) {
    const {
      className,
      id,
      name,
      required,
      disabled,
      label,
      "aria-label": ariaLabel,
      "aria-labelledby": ariaLabelledBy,
      description,
      "aria-describedby": ariaDescribedBy,
      onChange,
      onCancel,
      onClear,
      status,
      statusMessage,
      labels: labelsProp,
      defaultValue,
      value,
      filters,
      horizontalPlacement = "right",
      disablePortal = true,
      escapeWithReference = true,
      height = 350,
      filterContentProps,
      classes: classesProp,
      ...others
    } = uikitReactUtils.useDefaultProps("HvFilterGroup", props);
    const { classes, cx } = FilterGroup_styles.useClasses(classesProp);
    const [validationMessage] = useControlled.useControlled(statusMessage, "Required");
    const elementId = useUniqueId.useUniqueId(id);
    const labels = useLabels.useLabels(DEFAULT_LABELS, labelsProp);
    const hasLabel = label != null;
    const hasDescription = description != null;
    const canShowError = status !== void 0 || required;
    return /* @__PURE__ */ jsxRuntime.jsxs(
      FormElement.HvFormElement,
      {
        id,
        name,
        value,
        status,
        disabled,
        required,
        className: cx(classes.root, className),
        ...others,
        children: [
          (hasLabel || hasDescription) && /* @__PURE__ */ jsxRuntime.jsxs("div", { className: classes.labelContainer, children: [
            hasLabel && /* @__PURE__ */ jsxRuntime.jsx(
              Label.HvLabel,
              {
                showGutter: true,
                id: setId.setId(elementId, "label"),
                htmlFor: setId.setId(elementId, "input"),
                label,
                className: classes.label
              }
            ),
            hasDescription && /* @__PURE__ */ jsxRuntime.jsx(
              InfoMessage.HvInfoMessage,
              {
                id: setId.setId(elementId, "description"),
                className: classes.description,
                children: description
              }
            )
          ] }),
          /* @__PURE__ */ jsxRuntime.jsxs(
            FilterGroupContext.HvFilterGroupProvider,
            {
              defaultValue,
              value,
              filters,
              children: [
                /* @__PURE__ */ jsxRuntime.jsx(
                  FilterContent.HvFilterGroupContent,
                  {
                    ref,
                    id: elementId,
                    disabled,
                    disablePortal,
                    variableWidth: true,
                    placement: horizontalPlacement,
                    escapeWithReference,
                    "aria-label": ariaLabel,
                    "aria-labelledby": ariaLabelledBy,
                    status,
                    onChange,
                    onCancel,
                    onClear,
                    labels,
                    height,
                    ...filterContentProps
                  }
                ),
                canShowError && /* @__PURE__ */ jsxRuntime.jsx(
                  WarningText.HvWarningText,
                  {
                    id: setId.setId(elementId, "error"),
                    disableBorder: true,
                    className: classes.error,
                    children: validationMessage
                  }
                )
              ]
            }
          )
        ]
      }
    );
  }
);
exports.filterGroupClasses = FilterGroup_styles.staticClasses;
exports.HvFilterGroup = HvFilterGroup;
