// @hitachivantara/uikit-react-core/dist/cjs/Card/Card.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvCard", {
  root: {
    overflow: "visible",
    position: "relative",
    outline: `1px solid ${uikitStyles.theme.colors.atmo4}`,
    borderRadius: `0px 0px ${uikitStyles.theme.radii.round} ${uikitStyles.theme.radii.round}`,
    backgroundColor: "var(--bg-color)",
    "&:focus-visible": {
      ...focusUtils.outlineStyles
    },
    "&:focus": {
      outline: "none"
    }
  },
  selected: {
    outline: `1px solid ${uikitStyles.theme.colors.secondary}`,
    "&:hover": {
      outline: `1px solid ${uikitStyles.theme.colors.secondary}`
    },
    "&:focus": {
      outline: `1px solid ${uikitStyles.theme.colors.secondary}`
    }
  },
  selectable: {
    "&:hover": {
      outline: `1px solid ${uikitStyles.theme.colors.primary}`
    }
  },
  semanticContainer: {
    position: "relative",
    "& > *": {
      position: "absolute",
      zIndex: 1
    }
  },
  icon: {
    top: uikitStyles.theme.space.xs,
    right: uikitStyles.theme.space.xs
  },
  semanticBar: {
    backgroundColor: "var(--bar-color)",
    height: "var(--bar-height)",
    width: "100%",
    top: -1,
    right: 0
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
