// @hitachivantara/uikit-react-core/dist/cjs/VerticalNavigation/NavigationSlider/NavigationSlider.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses(
  "HvVerticalNavigationSlider",
  {
    root: {
      display: "flex",
      alignItems: "center",
      borderLeft: `4px solid transparent`,
      minHeight: "48px",
      color: "inherit",
      marginBottom: "8px",
      "& > button": {
        marginLeft: "auto"
      },
      "& .HvListItem-startAdornment .color0": {
        fill: "currentColor"
      }
    },
    listItemSelected: {
      background: uikitStyles.theme.colors.atmo3,
      borderLeft: `4px solid ${uikitStyles.theme.colors.secondary}`
    },
    listItemFocus: {
      background: uikitStyles.theme.colors.atmo3
    },
    forwardButton: {
      color: "inherit"
    },
    listContainer: {},
    listItemDisabled: {}
  }
);
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
