// @hitachivantara/uikit-react-core/dist/cjs/DotPagination/DotPagination.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const uikitReactIcons = require("@hitachivantara/uikit-react-icons");
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const helpers = require("../utils/helpers.cjs");
const DotPagination_styles = require("./DotPagination.styles.cjs");
const RadioGroup = require("../RadioGroup/RadioGroup.cjs");
const Radio = require("../Radio/Radio.cjs");
const getSelectorIcons = (radioIcon, radioCheckedIcon, classes) => {
  return {
    radio: React.cloneElement(radioIcon || /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.OtherStep, { width: 8, height: 8 }), {
      className: classes?.icon
    }),
    radioChecked: React.cloneElement(
      radioCheckedIcon || /* @__PURE__ */ jsxRuntime.jsx(uikitReactIcons.CurrentStep, { width: 8, height: 8 }),
      {
        className: classes?.icon
      }
    )
  };
};
const HvDotPagination = (props) => {
  const {
    className,
    classes: classesProp,
    unselectedIcon,
    selectedIcon,
    pages = 1,
    page = 0,
    onPageChange,
    getItemAriaLabel,
    ...others
  } = uikitReactUtils.useDefaultProps("HvDotPagination", props);
  const { classes, cx } = DotPagination_styles.useClasses(classesProp);
  const icons = getSelectorIcons(unselectedIcon, selectedIcon, classes);
  return /* @__PURE__ */ jsxRuntime.jsx(
    RadioGroup.HvRadioGroup,
    {
      className: cx(classes.root, className),
      classes: {
        horizontal: classes.horizontal
      },
      orientation: "horizontal",
      ...others,
      children: helpers.range(pages).map((i) => /* @__PURE__ */ jsxRuntime.jsx(
        Radio.HvRadio,
        {
          classes: {
            radio: classes.radio,
            root: classes.radioRoot
          },
          value: i,
          checked: page === i,
          onChange: (event) => onPageChange?.(event, i),
          icon: icons.radio,
          checkedIcon: icons.radioChecked,
          "aria-label": getItemAriaLabel?.(i)
        },
        i
      ))
    }
  );
};
exports.dotPaginationClasses = DotPagination_styles.staticClasses;
exports.HvDotPagination = HvDotPagination;
