// @hitachivantara/uikit-react-core/dist/esm/BreadCrumb/Page/Page.js
import { jsx } from "react/jsx-runtime";
import { useDefaultProps } from "@hitachivantara/uikit-react-utils";
import { useClasses } from "./Page.styles.js";
import { staticClasses } from "./Page.styles.js";
import { HvTypography } from "../../Typography/Typography.js";
import { HvOverflowTooltip } from "../../OverflowTooltip/OverflowTooltip.js";
const HvBreadCrumbPage = (props) => {
  const {
    component,
    onClick,
    elem,
    classes: classesProp
  } = useDefaultProps("HvBreadCrumbPage", props);
  const { classes, cx } = useClasses(classesProp);
  const { label, path, ...others } = elem;
  const handleClick = (event) => {
    event.preventDefault();
    onClick?.(event, elem);
  };
  return /* @__PURE__ */ jsx(
    HvTypography,
    {
      noWrap: true,
      variant: "label",
      component: component || "a",
      href: elem.path,
      onClick: onClick && handleClick,
      className: cx(classes.link, classes.label, classes.a),
      ...others,
      children: /* @__PURE__ */ jsx(HvOverflowTooltip, { data: elem.label })
    }
  );
};
export {
  HvBreadCrumbPage,
  staticClasses as breadCrumbPageClasses
};
