// @hitachivantara/uikit-react-core/dist/cjs/ListContainer/ListItem/ListItem.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const focusUtils = require("../../utils/focusUtils.cjs");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvListItem", {
  root: {
    ...uikitStyles.theme.typography.body,
    padding: 0,
    display: "block",
    height: "32px",
    lineHeight: "32px",
    listStyleType: "none",
    overflow: "hidden",
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    position: "relative",
    alignContent: "center",
    "&:not(:last-child)": {
      marginBottom: "8px"
    },
    "&$condensed": {
      marginBottom: 0
    },
    "&.HvIsFocused": {
      ...focusUtils.outlineStyles,
      backgroundColor: uikitStyles.theme.colors.atmo3,
      zIndex: 2
    }
  },
  focus: { backgroundColor: uikitStyles.theme.colors.atmo3, zIndex: 2 },
  startAdornment: {},
  endAdornment: {},
  gutters: {
    padding: `0 ${uikitStyles.theme.space.xs}`,
    "&$withStartAdornment": {
      paddingLeft: 0
    },
    "&$withEndAdornment": {
      paddingRight: 0
    }
  },
  condensed: {},
  interactive: {
    cursor: "pointer",
    "&:not($disabled):not($selected):hover": {
      backgroundColor: uikitStyles.theme.colors.containerBackgroundHover
    },
    "&$disabled": {
      cursor: "not-allowed"
    }
  },
  selected: { backgroundColor: uikitStyles.theme.colors.containerBackgroundHover },
  disabled: {
    color: uikitStyles.theme.colors.secondary_60,
    backgroundColor: uikitStyles.theme.colors.atmo3
  },
  withStartAdornment: {
    "& > div": {
      float: "left"
    },
    "& svg": {
      boxShadow: "none !important",
      outline: "none !important"
    },
    "$disabled > svg *.color0": {
      fill: uikitStyles.theme.colors.secondary_60
    }
  },
  withEndAdornment: {
    "& > div": { float: "right" },
    "& svg": {
      boxShadow: "none !important",
      outline: "none !important"
    },
    "$disabled > svg *.color0": {
      fill: uikitStyles.theme.colors.secondary_60
    }
  }
});
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
