// @hitachivantara/uikit-react-core/dist/cjs/SimpleGrid/SimpleGrid.styles.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const uikitReactUtils = require("@hitachivantara/uikit-react-utils");
const uikitStyles = require("@hitachivantara/uikit-styles");
const { staticClasses, useClasses } = uikitReactUtils.createClasses("HvSimpleGrid", {
  root: {
    display: "grid",
    boxSizing: "border-box",
    gridTemplateColumns: `repeat(var(--cols, 1), minmax(0, 1fr))`
  }
});
function getSize(size) {
  return size || Number(uikitStyles.theme.breakpoints.values.md);
}
function getSortedBreakpoints(breakpoints) {
  if (breakpoints.length === 0) {
    return breakpoints;
  }
  const property = "maxWidth" in breakpoints[0] ? "maxWidth" : "minWidth";
  const sorted = [...breakpoints].sort(
    (a, b) => getSize(b[property]) - getSize(a[property])
  );
  return property === "minWidth" ? sorted.reverse() : sorted;
}
const getContainerStyle = (breakpoints, spacing = "sm", cols = 1) => {
  return {
    // TODO: review/document precedence of cols/spacing vs breakpoints[cols/spacing]
    "--cols": cols,
    gap: uikitStyles.theme.space[spacing],
    ...breakpoints && getSortedBreakpoints(breakpoints).reduce((acc, breakpoint) => {
      const property = "maxWidth" in breakpoint ? "max-width" : "min-width";
      const breakpointSize = getSize(
        property === "max-width" ? breakpoint.maxWidth : breakpoint.minWidth
      );
      acc[`@media (${property}: ${breakpointSize}px)`] = {
        ["--cols"]: breakpoint.cols,
        gap: uikitStyles.theme.space[breakpoint.spacing || spacing]
      };
      return acc;
    }, {})
  };
};
exports.getContainerStyle = getContainerStyle;
exports.staticClasses = staticClasses;
exports.useClasses = useClasses;
